---
title: "Gene coexpression network inference with BioNERO"
author: 
- name: Fabricio Almeida-Silva
  affiliation: Universidade Estadual do Norte Fluminense Darcy Ribeiro, RJ, Brazil
- name: Thiago Motta Venancio
  affiliation: Universidade Estadual do Norte Fluminense Darcy Ribeiro, RJ, Brazil
output: 
  BiocStyle::html_document:
    toc: true
    highlight: pygments
    number_sections: yes
vignette: >
  %\VignetteIndexEntry{Gene coexpression network inference with BioNERO}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  message = FALSE,
  warning = FALSE,
  cache = TRUE,
  fig.align = 'center',
  fig.width = 6,
  fig.height = 5
)
```

# Introduction

To date, several packages have been developed to infer gene coexpression networks and gene regulatory networks from expression data, such as WGCNA, CEMiTool, petal, GENIE3 and minet. However, network inference and analysis is a non-trivial task that requires solid statistical background, especially for data preprocessing and proper interpretation of results. Because of that, inexperienced researchers often struggle to choose the most suitable algorithms for data preprocessing and analysis based on their particular objectives. Besides, different packages are required for each step of a standard network analysis, and their distinct syntaxes can hinder interoperability between packages, particularly for non-advanced R programmers. Here, we have developed an all-in-one R package that uses state-of-the-art algorithms to facilitate the workflow of biological network analysis, from data acquisition to analysis and interpretation. This will likely accelerate network analysis pipelines and advance systems biology research.

# Installation

```{r eval=FALSE}
if(!requireNamespace('BiocManager', quietly = TRUE))
  install.packages('BiocManager')

BiocManager::install("BioNERO")
```

```{r}
# Load package after installation
library(BioNERO)
```

# Data loading and pre-processing

For this tutorial, we will use gene expression data (TPM-normalized) of soybean seeds downloaded from the **Soybean Expression Atlas** [(Machado *et al*, 2020)](https://doi.org/10.1111/tpj.14850). For simplicity, we will use a subset of a single study (SRA BioProject: PRJNA197251) by [Danzer et al, 2015](https://doi.org/10.1104/pp.15.00432). The data set is stored as a SummarizedExperiment object for convenience. Further information on the example data set can be accessed with `?se.seed`.

**NOTE:** In case you have many tab-separated expression tables in a directory, `BioNERO` has a helper function named `dfs2one()` to load all these files and combine them into a single data frame. For that, make sure all files have gene IDs in the first column and sample names in the other columns. Then, you could use:

```{r eval=FALSE}
exp <- dfs2one(mypath="path/to/directory/")
```

First of all, let's create an expression data frame and a metadata data frame to be used in further analyses.

```{r data_loading_soybean}
library(SummarizedExperiment)
data(se.seed)

# Take a quick look at the data
se.seed
colData(se.seed)

# Create expression data frame
exp <- assay(se.seed)

# Create metadata data frame
metadata <- data.frame(Sample = rownames(colData(se.seed)),
                       Tissue = colData(se.seed)$Part)

```


## Step-by-step data pre-processing

This section is suitable for users who want to have more control of their data analysis, since they can inspect the data set after each pre-processing step and analyze how different options to the arguments would affect the expression data. You can skip to the next section (**Automatic, one-step data preprocessing**) for a quick start.

**Step 1:** Removing missing values. By default, `remove_na()` will replace NAs with 0. Users can also replace NAs with the mean of each row.

```{r remove_na}
exp_filt <- remove_na(exp)
```

**Step 2:** Removing non-expressed genes. Here, for faster network reconstruction, we will remove every gene whose median value is below 5. The function's default for `min_exp` is 1. For other options, see `?remove_nonexp`. 

```{r remove_nonexp}
exp_filt <- remove_nonexp(exp_filt, method="median", min_exp = 5)
```

**Step 3 (optional):** Filtering genes by variance. It is reasonable to remove genes whose expression values do not vary much across samples, since we often want to find genes that are more or less expressed in particular conditions. Here, we will keep only the top 3000 most variable genes. Users can also filter by percentile (e.g., the top 10% most variable genes).

```{r filter_by_variance}
exp_filt <- filter_by_variance(exp_filt, n=3000)
```

**Step 4:** Removing outlying samples. There are several methods to remove outliers. We have implemented the Z.K (standardized connectivity) method in `ZKfiltering()`, which is a network-based approach to remove outlying samples. This method has proven to be more suitable for network analysis, since it can remove outliers that other methods (such as hierarchical clustering) cannot identify. By default, BioNERO considers all samples with Zk < 2 as outliers, but this parameter is flexible if users want to change it.

```{r ZKfiltering}
dim(exp_filt)
exp_filt <- ZKfiltering(exp_filt, cor_method = "pearson")
```

**Step 5:** Adjusting for confounding artifacts. This is an important step to avoid spurious correlations resulting from confounders. The method was described by [Parsana *et al*, 2019](https://doi.org/10.1186/s13059-019-1700-9), who developed a principal component (PC)-based correction for confounders. After correction, the expression data are quantile normalized so that every gene follows an approximate normal distribution.

```{r PC_correction}
exp_filt <- PC_correction(exp_filt)
```


## Automatic, one-step data preprocessing

Alternatively, users can preprocess their data with a single function. The function `exp_preprocess()` is a wrapper for the functions `remove_na()`, `remove_nonexp()`, `filter_by_variance()`, `ZKfiltering()` and `PC_correction()`. The arguments passed to `exp_preprocess()` will be used by each of these functions to generate a filtered expression data in a single step.

```{r exp_preprocess}
final_exp <- exp_preprocess(exp, min_exp = 5, variance_filter = TRUE, n=3000)
dim(exp_filt) == dim(final_exp)
```

```{r include=FALSE}
# This object is no longer necessary
rm(exp_filt)
```


# Exploratory data analysis

`BioNERO` includes some functions for easy data exploration. These functions were created to avoid having to type code chunks that, although small, will be used many times. Our whole idea here is to make the user experience with biological network analysis as easy and simple as possible.

**Plotting heatmaps:** the function `plot_heatmap()` plots heatmaps of correlations between samples or gene expression in a single line. Users can use their preferred RColorBrewer's palette, hide/show gene names, and activate/deactivate clustering for rows and/or columns.

```{r plot_heatmap, fig.wide=TRUE, fig.align='center'}
# Simple heatmap of sample correlations
plot_heatmap(final_exp, type = "samplecor")

# Add sample descriptions
plot_heatmap(final_exp, type = "samplecor", col_metadata = metadata)

# Simple heatmap of gene expression
plot_heatmap(final_exp, type="expr")

# Add sample descriptions
plot_heatmap(final_exp, type="expr", col_metadata = metadata)

# Remove sample clustering
plot_heatmap(final_exp, type="expr", col_metadata = metadata, cluster_cols=FALSE)

# Remove gene clustering
plot_heatmap(final_exp, type="expr", col_metadata = metadata, cluster_rows=FALSE)
```


**Principal component analysis (PCA):** the function `plot_PCA()` performs a PCA and plots PC1 vs PC2 (by default), as well the percentage of variance explained by each PC. The users can also choose to plot PC1 vs PC3 or PC2 vs PC3.

```{r pcaplot, cache=FALSE}
# Simple PCA plot
pca <- plot_PCA(final_exp, metadata, size=4)
pca 
```

Other functions exist for exploratory purposes, such as `plot_expression_profile()` and `plot_ngenes_per_module()`, but they were made to be used after gene coexpression network reconstruction or for a set of genes of interest.


# Gene coexpression network reconstruction

Now that we have our filtered and normalized expression data, we can reconstruct a gene coexpression network (GCN). First of all, we need to identify the most suitable $\beta$ power that makes the network satisfy the scale-free topology. We do that with the function `SFT_fit()`. Correlation values are raised to a power $\beta$ in order to amplify the distances between correlations and, hence, to make the module detection algorithm more powerful. The higher the value of $\beta$, the closer to the scale-free topology the network is. However, a very high $\beta$ power reduces mean connectivity, which is not desired. To solve this trade-off, we usually pick the lowest $\beta$ power above a certain threshold (by default in `SFT_fit()`, 0.8). This makes the network close to the scale-free topology without reducing dramatically the mean connectivity.

```{r sft_fit}
sft <- SFT_fit(final_exp, net_type="signed hybrid", cor_method="pearson")
sft$power
power <- sft$power
```

As we can see, the optimal power is `r power`. If the users want to visually inspect the simulation of different $\beta$ values, the function `SFT_fit()` automatically saves a ggplot object in the second element of the resulting list. To visualize it, you simply have to access the plot.

```{r}
sft$plot
```

Here comes the actual network inference! Now, we can use the power calculated by `SFT_fit()` to reconstruct the GCN. The function `exp2net()` reconstructs a GCN and outputs a list of 5 elements, each of which can be used by other functions in the analysis pipeline.

```{r exp2net}
net <- exp2net(final_exp, net_type="signed hybrid", SFTpower=power, cor_method="pearson")
names(net)
str(net)
```

# Gene coexpression network analysis

Now that we have our coexpression network, we can start exploring some of its properties. 

## Assessing module stability

The function `module_stability()` allows users to check if the identified coexpression modules are stable (i.e., if they can resist removal of a particular sample). This function will sample the data set and rerun the module detection algorithm *n* times (default: 30) and return a PDF figure displaying a gene dendrogram and colors representing modules identified in each run. By looking at the figure, you can detect if a particular module is only found in a very small fraction of the runs, which suggests instability.

**NOTE:** The calculations performed by this function may take a long time depending on the size of your expression data. Use it only if you have some reason to suspect that the modules are highly dependent on a particular set of samples (or if you are too skeptical to just keep going).

```{r module_stability, eval=FALSE}
module_stability(final_exp, net, nRuns=15)
```

```{r mod_stability_plot, echo=FALSE, eval=FALSE}
knitr::include_graphics("module_stability.pdf")
```

## Module-trait associations

The function `module_trait_cor()` can be used to calculate module-trait correlations. In this case, tissues will be considered traits. This analysis is useful to identify modules that are positively or negatively correlated with particular traits, which means that their gene expression levels go up or down in these conditions. By treating tissues as traits, we want to identify groups of genes whose expression levels are inhibited or enhanced in particular tissues. Alternatively, one can also use continuous variables (e.g, metabolite content, protein concentration, height) or discrete variables (e.g., disease index) as traits.

```{r moddtraitcor, fig.width=5, fig.height=5}
module_trait_cor(exp=final_exp, metadata=metadata, MEs=net$MEs, 
                 cor_method="pearson")
```

The function `module_trait_cor()` also allows for plot customization. For instance:

```{r modtraitcor_transpose, fig.width=8, fig.height=3}
# Transpose the matrix
module_trait_cor(exp=final_exp, metadata=metadata, MEs=net$MEs, 
                 cor_method="pearson", transpose = TRUE)
```

```{r mod_trait_cor_pal, fig.width=5, fig.height=5}
# Change the palette (based on RColorBrewer's palette)
module_trait_cor(exp=final_exp, metadata=metadata, MEs=net$MEs,
                 cor_method="pearson", palette="PRGn")
```


## Enrichment analysis

After identifying modules that are inhibited or enhanced in particular tissues, users would likely want to find to which biological processes (e.g., GO biological process) or pathways (e.g., Reactome, KEGG, MapMan) these genes are related. This can be done with enrichment analyses, which can uncover terms that are found more than expected by chance in a module.

The easiest way to accomplish this is to use the function `module_enrichment()`, which performs enrichment analysis for all modules at once. To illustrate it, we will scan coexpression modules for enriched conserved protein domains using all genes in the network as background. The Interpro annotation will be directly loaded from the **[PLAZA 4.0 database](https://bioinformatics.psb.ugent.be/plaza/versions/plaza_v4_dicots/download/index)**.

```{r module_enrichment}
# Enrichment analysis for conserved protein domains (Interpro)
interpro_annot <- as.data.frame(readr::read_tsv("ftp://ftp.psb.ugent.be/pub/plaza/plaza_public_dicots_04/InterPro/interpro.gma.csv.gz", skip = 8))[, c(1,4)] 
head(interpro_annot)
interpro_enrichment <- module_enrichment(net = net, 
                                         background_genes = rownames(final_exp),
                                         annotation = interpro_annot)
interpro_enrichment
```

As we can see, the module **yellow** is enriched in genes encoding ethylene-responsive transcription factors (ERF/AP2). We could get the same result with the function `enrichment_analysis()`, which performs enrichment analysis for a user-defined gene set instead of all modules.

```{r enrichment_analysis}
# Get character vector of genes in the module yellow
yellow <- net$genes_and_modules$Genes[net$genes_and_modules$Modules == "yellow"]

yellow_enrichment <- enrichment_analysis(genes = yellow,
                                         background_genes = rownames(final_exp),
                                         annotation = interpro_annot)
yellow_enrichment
```

## Hub gene identification

Hub genes are often identified using two different metrics: **module membership (MM)** (i.e., correlation of a gene to its module eigengene) and **degree** (i.e., sum of connections weights of a gene to all other genes in the module). Some researchers consider the top 10% genes with the highest degree as hubs, while others consider those with MM > 0.8. To avoid false positives, `BioNERO`'s algorithm combines both metrics and defines hub genes as the top 10% genes with highest degree that have MM > 0.8. Hubs can be identified with the function `get_hubs()`.

```{r}
hubs <- get_hubs(final_exp, net)
head(hubs)
```
## Extracting subgraphs

Subgraph extraction can be particularly useful to visualize specific modules, and it can be done with the function `get_edge_list()`. The function returns the subgraph as an edge list. Users can also extract an edge list for a particular gene set instead of a module.

```{r}
# Extract all edges of the module 'yellow'
yellow_edges <- get_edge_list(net, module="yellow")
head(yellow_edges)
dim(yellow_edges)
```

The function `get_edge_list()` returns a fully connected subgraph for the specified module or gene set. However, one would almost always want to filter out very weak connections. This can be accomplished by setting the argument `filter` to `TRUE`, which will remove edges based on optimal scale-free topology fit (default), p-value, Z-score, or an arbitrary minimum correlation coefficient. For more details details, check `?get_edge_list()`.

```{r}
# Remove edges based on optimal scale-free topology fit
yellow_edges_filtered <- get_edge_list(net, module="yellow",
                                       filter=TRUE)
dim(yellow_edges_filtered)

# Remove edges based on p-value
yellow_edges_filtered <- get_edge_list(net, module="yellow",
                                       filter=TRUE, method="pvalue",
                                       nSamples = ncol(final_exp))
dim(yellow_edges_filtered)
```

## Network visualization

As we now have an edge list for the module **yellow**, let's visualize it with the function `plot_gcn()`. By default, this function only labels the top 5 hubs (or less if there are less than 5 hubs). However, this can be customized according to the user's preference (see `?plot_gcn` for more information). For instance:

```{r}
# Network plot with labels for top hubs only
plot_gcn(edgelist_gcn = yellow_edges_filtered, 
         net = net,
         modulename = "yellow", 
         hubs = hubs)

# With labels for all genes
plot_gcn(edgelist_gcn = yellow_edges_filtered, 
         net = net,
         modulename = "yellow", 
         hubs = hubs,
         show_labels = "all")
```

Networks can also be visualized interactively by setting `interactive = TRUE` in `plot_gcn`.

```{r}
plot_gcn(edgelist_gcn = yellow_edges_filtered, 
         net = net,
         modulename = "yellow", 
         hubs = hubs,
         interactive = TRUE)
```


## Network statistics

Finally, the function `net_stats()` can be used to calculate the main network statistics (or properties, or indices), namely: *connectivity*, *scaled connectivity*, *clustering coefficient*, *maximum adjacency ratio*, *density*, *centralization*, *heterogeneity*, *number of cliques*, *diameter*, *betweenness* (optional), *closeness* (optional).

**NOTE:** Depending on your system capacities and network size, this may take a very long time. Hence, if you are willing to calculate network statistics for you data set, grab a cup of coffee, because the waiting will be long.

# Session information

This vignette was created under the following conditions:

```{r}
sessionInfo()
```










