<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Preprocess expression data for network reconstruction"><title>Preprocess expression data for network reconstruction — exp_preprocess • BioNERO</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.6/font.css" rel="stylesheet"><link href="../deps/Source_Code_Pro-0.4.6/font.css" rel="stylesheet"><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Preprocess expression data for network reconstruction — exp_preprocess"><meta property="og:description" content="Preprocess expression data for network reconstruction"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-#0092ac"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">BioNERO</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.9.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/vignette_01_GCN_inference.html">Gene coexpression network inference</a>
    <a class="dropdown-item" href="../articles/vignette_02_GRN_inference.html">Gene regulatory network inference</a>
    <a class="dropdown-item" href="../articles/vignette_03_network_comparison.html">Network comparison: consensus modules and module preservation</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/almeidasilvaf/BioNERO/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Preprocess expression data for network reconstruction</h1>
      <small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/BioNERO/blob/HEAD/R/data_preprocess.R" class="external-link"><code>R/data_preprocess.R</code></a></small>
      <div class="d-none name"><code>exp_preprocess.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Preprocess expression data for network reconstruction</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">exp_preprocess</span><span class="op">(</span></span>
<span>  <span class="va">exp</span>,</span>
<span>  NA_rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  replaceby <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  Zk_filtering <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  zk <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>,</span>
<span>  cor_method <span class="op">=</span> <span class="st">"spearman"</span>,</span>
<span>  remove_nonexpressed <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  min_exp <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  min_percentage_samples <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  remove_confounders <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  variance_filter <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  n <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  percentile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  vstransform <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>exp</dt>
<dd><p>A gene expression data frame with genes in row names
and samples in column names or a `SummarizedExperiment` object.</p></dd>


<dt>NA_rm</dt>
<dd><p>Logical. It specifies whether to remove missing values
from the expression data frame or not. Default = TRUE.</p></dd>


<dt>replaceby</dt>
<dd><p>If NA_rm is TRUE, what to use instead of NAs.
One of 0 or 'mean'. Default is 0.</p></dd>


<dt>Zk_filtering</dt>
<dd><p>Logical. It specifies whether to filter outlying samples
by Zk or not. Default: TRUE.</p></dd>


<dt>zk</dt>
<dd><p>If Zk_filtering is TRUE, the standardized connectivity threshold.
Samples below this threshold will be considered outliers. Default is -2.</p></dd>


<dt>cor_method</dt>
<dd><p>If Zk_filtering is TRUE, the correlation method to use.
One of 'spearman', 'bicor', or 'pearson'. Default is 'spearman'.</p></dd>


<dt>remove_nonexpressed</dt>
<dd><p>Logical. It specifies whether non-expressed genes
should be removed or not. Default is TRUE.</p></dd>


<dt>method</dt>
<dd><p>If remove_nonexpressed is TRUE, the criterion to filter
non-expressed genes out. One of "mean", "median", "percentage",
or "allsamples". Default is 'median'.</p></dd>


<dt>min_exp</dt>
<dd><p>If method is 'mean', 'median', or 'allsamples',
the minimum value for a gene to be considered expressed.
If method is 'percentage', the minimum value each gene must have in at least
n percent of samples to be considered expressed.</p></dd>


<dt>min_percentage_samples</dt>
<dd><p>If method is 'percentage', expressed genes
must have expression &gt;= min_exp in at least this percentage.
Values must range from 0 to 1. Default = 0.25.</p></dd>


<dt>remove_confounders</dt>
<dd><p>Logical. If TRUE, it removes principal components
that add noise to the data.</p></dd>


<dt>variance_filter</dt>
<dd><p>Logical. If TRUE, it will filter genes by variance.
Default is FALSE.</p></dd>


<dt>n</dt>
<dd><p>If variance_filter is TRUE, the number of
most variable genes to keep.</p></dd>


<dt>percentile</dt>
<dd><p>If variance_filter is TRUE, the percentage of
most variable genes to keep.</p></dd>


<dt>vstransform</dt>
<dd><p>Logical indicating if data should be
variance stabilizing transformed. This parameter can only be set to TRUE
if data is a matrix of raw read counts.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Processed gene expression data frame with gene IDs in row names
and sample names in column names or `SummarizedExperiment` object.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Love, M. I., Huber, W., &amp; Anders, S. (2014). Moderated estimation of
fold change and dispersion for RNA-seq data with
DESeq2. Genome biology, 15(12), 1-21.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/DESeq2/man/varianceStabilizingTransformation.html" class="external-link">varianceStabilizingTransformation</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Fabricio Almeida-Silva</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">zma.se</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu">exp_preprocess</span><span class="op">(</span><span class="va">zma.se</span>, variance_filter<span class="op">=</span><span class="cn">TRUE</span>, n<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of removed samples: 1</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Fabricio Almeida-Silva, Thiago Venancio.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

