<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BioNERO: Easy biological network reconstruction and analysis â€¢ BioNERO</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BioNERO: Easy biological network reconstruction and analysis">
<meta property="og:description" content="BioNERO">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioNERO</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/BioNERO.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/almeidasilvaf/BioNERO/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BioNERO_files/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="BioNERO_files/jquery-1.11.0/jquery.min.js"></script><script src="BioNERO_files/detect-resize-0.5.3/jquery.resize.js"></script><link href="BioNERO_files/jquery-ui-1.11.4/jquery-ui.min.css" rel="stylesheet">
<script src="BioNERO_files/jquery-ui-1.11.4/jquery-ui.min.js"></script><script src="BioNERO_files/d3-3.5.2/d3.min.js"></script><script src="BioNERO_files/vega-1.4.3/vega.min.js"></script><script src="BioNERO_files/lodash-2.2.1/lodash.min.js"></script><script>var lodash = _.noConflict();</script><link href="BioNERO_files/ggvis-0.4.4.9000/css/ggvis.css" rel="stylesheet">
<script src="BioNERO_files/ggvis-0.4.4.9000/js/ggvis.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BioNERO: Easy biological network reconstruction and analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/BioNERO/blob/master/vignettes/BioNERO.Rmd"><code>vignettes/BioNERO.Rmd</code></a></small>
      <div class="hidden name"><code>BioNERO.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/almeidasilvaf/BioNERO">BioNERO</a></span>)
</pre></div>
<div id="data-loading-and-pre-processing" class="section level2">
<h2 class="hasAnchor">
<a href="#data-loading-and-pre-processing" class="anchor"></a>1 Data loading and pre-processing</h2>
<p>For this tutorial, we will use soybean gene expression data in TPM, which was downloaded from the <strong>Soybean Expression Atlas</strong> <a href="https://doi.org/10.1111/tpj.14850">(Machado <em>et al</em>, 2020)</a>. This expression resource is very big, and analyzing all the available RNA-seq samples (n=1298) would be very time-consuming. For simplicity, we will use only a subset of the data base. The subset we will use comprises RNA-seq data generated by <a href="https://doi.org/10.1111/j.1365-313X.2010.04222.x">Libault <em>et al</em>, 2010</a> and <a href="https://doi.org/10.1186/1471-2229-10-160">Severin <em>et al</em>, 2010</a>.</p>
<p>Each of the two TPM expression table is a different tab-delimited file. The first column contains gene IDs and the other columns contain the expression values for each sample. First, we will read both files to R and combine them into a single data frame.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dfs2one.html">dfs2one</a></span>(mypath = <span class="st">"./TPM_data"</span>)
<span class="kw">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span>(<span class="st">"metadata.txt"</span>, header=<span class="kw">T</span>, sep=<span class="st">"\t"</span>, stringsAsFactors = <span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">exp</span>)
<span class="co">## [1] 56044    28</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">exp</span>)
<span class="co">##                 SAMN00009684 SAMN00009685 SAMN00009686 SAMN00009687</span>
<span class="co">## Glyma.01G000100     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000200     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000300     4.446539     3.105494     3.125155     3.470989</span>
<span class="co">## Glyma.01G000400     1.323173     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000500     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000600     3.286091     3.118817     2.473931     2.790083</span>
<span class="co">##                 SAMN00009688 SAMN00009689 SAMN00009690 SAMN00009691</span>
<span class="co">## Glyma.01G000100     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000200     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000300     4.263587     3.347583     3.678279     6.868057</span>
<span class="co">## Glyma.01G000400     5.381255     1.270567     2.115066     2.944164</span>
<span class="co">## Glyma.01G000500     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000600     4.113677     2.045503     6.997648     4.944498</span>
<span class="co">##                 SAMN00009692 SAMN00009693 SAMN00009694 SAMN00009695</span>
<span class="co">## Glyma.01G000100     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000200     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000300     7.594278     2.444403     4.689396     3.067142</span>
<span class="co">## Glyma.01G000400     2.115948     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000500     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000600     0.356737     1.073550     1.875560     0.000000</span>
<span class="co">##                 SAMN00009696 SAMN00009697 SAMN02197688 SAMN02197689</span>
<span class="co">## Glyma.01G000100     0.000000     0.000000    28.658333     0.000000</span>
<span class="co">## Glyma.01G000200     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000300     6.685717     4.667017     8.875157     4.915844</span>
<span class="co">## Glyma.01G000400     4.219975     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000500     1.025139     0.000000     0.000000     4.021842</span>
<span class="co">## Glyma.01G000600     7.141623     3.121840     5.558067     7.048468</span>
<span class="co">##                 SAMN02197690 SAMN02197691 SAMN02197692 SAMN02197693</span>
<span class="co">## Glyma.01G000100     0.000000     0.000000     9.461721     0.000000</span>
<span class="co">## Glyma.01G000200     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000300     5.521244     6.410473     6.021127     6.410473</span>
<span class="co">## Glyma.01G000400     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000500     2.381000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000600     5.142537    11.833758     5.998980    11.833758</span>
<span class="co">##                 SAMN02197694 SAMN02197696 SAMN02197697 SAMN02197698</span>
<span class="co">## Glyma.01G000100     0.000000    10.047276     4.202899     0.000000</span>
<span class="co">## Glyma.01G000200     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000300     5.890313     6.437370     5.101789     7.128838</span>
<span class="co">## Glyma.01G000400     0.000000     3.253933     1.499856     5.291486</span>
<span class="co">## Glyma.01G000500     0.000000     0.000000     0.568418     1.094704</span>
<span class="co">## Glyma.01G000600     6.203266     8.285732     5.096749     4.928969</span>
<span class="co">##                 SAMN02197699 SAMN02197700 SAMN02197701 SAMN02197702</span>
<span class="co">## Glyma.01G000100     0.000000     0.000000     0.000000     2.233557</span>
<span class="co">## Glyma.01G000200     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000300     2.848676     3.202543     3.202543     3.415944</span>
<span class="co">## Glyma.01G000400     0.000000     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000500     1.018565     0.000000     0.000000     0.000000</span>
<span class="co">## Glyma.01G000600     1.091161     1.012614     1.012614     3.286954</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">metadata</span>)
<span class="co">##         Sample Tissue</span>
<span class="co">## 1 SAMN00009684   root</span>
<span class="co">## 2 SAMN00009685   root</span>
<span class="co">## 3 SAMN00009686   root</span>
<span class="co">## 4 SAMN00009687   root</span>
<span class="co">## 5 SAMN00009688   root</span>
<span class="co">## 6 SAMN00009689   root</span>
</pre></div>
<p>The combined data frame contains 56044 genes and 28 samples.</p>
<div id="step-by-step-data-pre-processing" class="section level3">
<h3 class="hasAnchor">
<a href="#step-by-step-data-pre-processing" class="anchor"></a>1.1 Step-by-step data pre-processing</h3>
<p>This section is suitable for users who want to have more control of their data analysis, since they can inspect the data set after each pre-processing step and analyze how different options to the arguments would affect the expression data.</p>
<p><strong>Step 1:</strong> Removing missing values. By default, <code><a href="../reference/remove_na.html">remove_na()</a></code> will replace NAs with 0. Users can also replace NAs with the mean of each row.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_na.html">remove_na</a></span>(<span class="kw">exp</span>)
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">exp</span>)) <span class="co"># checking if NAs have been removed</span>
<span class="co">## [1] 32676</span>
</pre></div>
<p><strong>Step 2:</strong> Removing non-expressed genes. Here, we will remove every gene whose median value is below 1. For other options, see <code><a href="../reference/remove_nonexp.html">?remove_nonexp</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_nonexp.html">remove_nonexp</a></span>(<span class="kw">exp_filt</span>, method=<span class="st">"median"</span>, min_exp = <span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">exp</span>)
<span class="co">## [1] 56044    28</span>
</pre></div>
<p>After step 2, the number of genes reduced to 56044.</p>
<p><strong>Step 3 (optional):</strong> Filtering genes by variance. It is reasonable to remove genes whose expression values do not vary much across samples, since we often want to find genes that are more or less expressed in particular conditions. Here, we will keep only the top 4000 most variable genes for simplicity. Users can also filter by percentile (e.g.Â the top 10% most variable genes).</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_by_variance.html">filter_by_variance</a></span>(<span class="kw">exp_filt</span>, n=<span class="fl">4000</span>)
</pre></div>
<p><strong>Step 4:</strong> Removing outlying samples. There are several methods to remove outliers. We have implemented the Zk (standardized connectivity) method, which is a network-based approach to remove outlying samples. This method has proven to be more suitable for network analysis, since it can remove outliers that other methods (such as hierarchical clustering) cannot identify. By default, BioNERO considers all samples with Zk &lt; 2 as outliers, but this parameter is flexible if users want to change it.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">exp</span>)
<span class="co">## [1] 56044    28</span>
<span class="kw">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ZKfiltering.html">ZKfiltering</a></span>(<span class="kw">exp_filt</span>, cor_method = <span class="st">"pearson"</span>)
<span class="co">## [1] "Number of samples that were removed: 1"</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">exp</span>)
<span class="co">## [1] 56044    28</span>
</pre></div>
<p><strong>Step 5:</strong> Adjusting for confounding artifacts. This is an important step to avoid spurious correlations resulting from confounders. The method was described by <a href="https://doi.org/10.1186/s13059-019-1700-9">Parsana <em>et al</em>, 2019</a>, who developed a principal component (PC)-based correction for confounders. After correction, the expression data is quantile normalized so that every gene follows an approximate normal distribution.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PC_correction.html">PC_correction</a></span>(<span class="kw">exp_filt</span>)
<span class="co">## [1] "Calculating number of PCs to be removed..."</span>
<span class="co">## [1] "Number of PCs estimated to be removed: 3"</span>
<span class="co">## [1] "Removing PCs that contribute to noise..."</span>
</pre></div>
<p>Letâ€™s check if genes really follow an approximate normal distribution.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mfrow=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">2</span>))
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(<span class="kw">exp_filt</span>[<span class="fl">1</span>,], main=<span class="st">"Histogram of gene 1"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(<span class="kw">exp_filt</span>[<span class="fl">2</span>,], main=<span class="st">"Histogram of gene 2"</span>)
</pre></div>
<p><img src="BioNERO_files/figure-html/unnamed-chunk-8-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div id="automatic-one-step-data-pre-processing" class="section level3">
<h3 class="hasAnchor">
<a href="#automatic-one-step-data-pre-processing" class="anchor"></a>1.2 Automatic, one-step data pre-processing</h3>
<p>Alternatively, users can pre-process their data with a single function. The function <code>exp_preprocess</code> is a wrapper for the functions <code>remove_na</code>, <code>remove_nonexp</code>, <code>filter_by_variance</code>, <code>ZKfiltering</code> and <code>PC_correction</code>. The arguments passed to <code>exp_preprocess</code> will be used by each of these functions to generate a filtered expression data in a single step.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw">exp_filt2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exp_preprocess.html">exp_preprocess</a></span>(<span class="kw">exp</span>, variance_filter = <span class="fl">TRUE</span>, n=<span class="fl">4000</span>)
<span class="co">## [1] "Number of samples that were removed: 1"</span>
<span class="co">## [1] "Calculating number of PCs to be removed..."</span>
<span class="co">## [1] "Number of PCs estimated to be removed: 3"</span>
<span class="co">## [1] "Removing PCs that contribute to noise..."</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">exp_filt</span>) <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw">exp_filt2</span>)
<span class="co">## [1] TRUE TRUE</span>
<span class="kw">finalexp</span> <span class="op">&lt;-</span> <span class="kw">exp_filt2</span> <span class="co"># save the filtered expression data a single object</span>
</pre></div>
</div>
</div>
<div id="exploratory-data-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#exploratory-data-analysis" class="anchor"></a>2 Exploratory data analysis</h2>
<p>BioNERO includes some functions for easy data exploration. These functions were created to avoid having to type code chunks that, although small, will be used many times. Our whole idea here is to make the user experience with biological network analysis as easy and simple as possible.</p>
<p><strong>Plotting heatmaps:</strong> the function <code>plot_heatmap</code> plots heatmaps of correlations between samples or gene expression in a single line. Users can use their preferred RColorBrewerâ€™s palette, hide/show gene names, and activate/deactivate clustering for rows and/or columns.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co"># Simple heatmap of sample correlations</span>
<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(<span class="kw">finalexp</span>, type = <span class="st">"samplecor"</span>)
</pre></div>
<p><img src="BioNERO_files/figure-html/unnamed-chunk-11-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb11"><pre class="downlit">

<span class="co"># Add sample descriptions</span>
<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(<span class="kw">finalexp</span>, type = <span class="st">"samplecor"</span>, col_metadata = <span class="kw">metadata</span>)
</pre></div>
<p><img src="BioNERO_files/figure-html/unnamed-chunk-11-2.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb12"><pre class="downlit">

<span class="co"># Simple heatmap of gene expression</span>
<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(<span class="kw">finalexp</span>, type=<span class="st">"expr"</span>)
</pre></div>
<p><img src="BioNERO_files/figure-html/unnamed-chunk-11-3.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb13"><pre class="downlit">

<span class="co"># Add sample descriptions</span>
<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(<span class="kw">finalexp</span>, type=<span class="st">"expr"</span>, col_metadata = <span class="kw">metadata</span>)
</pre></div>
<p><img src="BioNERO_files/figure-html/unnamed-chunk-11-4.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb14"><pre class="downlit">

<span class="co"># Remove sample clustering</span>
<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(<span class="kw">finalexp</span>, type=<span class="st">"expr"</span>, col_metadata = <span class="kw">metadata</span>, cluster_cols=<span class="fl">FALSE</span>)
</pre></div>
<p><img src="BioNERO_files/figure-html/unnamed-chunk-11-5.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb15"><pre class="downlit">

<span class="co"># Remove gene clustering</span>
<span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span>(<span class="kw">finalexp</span>, type=<span class="st">"expr"</span>, col_metadata = <span class="kw">metadata</span>, cluster_rows=<span class="fl">FALSE</span>)
</pre></div>
<p><img src="BioNERO_files/figure-html/unnamed-chunk-11-6.png" width="576" style="display: block; margin: auto;"></p>
<p><strong>Principal component analysis (PCA):</strong> the function <code>plot_PCA</code> performs a PCA and plots PC1 vs PC2 (by default), as well the percentage of variance explained by each PC. The users can also choose to plot PC1 vs PC3 or PC2 vs PC3. Besides, if <code>interactive = TRUE</code>, the function will display an interactive PCA plot.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="co"># Simple PCA plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/almeidasilvaf/BioNERO">BioNERO</a></span>)
<span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span>(<span class="kw">finalexp</span>, <span class="kw">metadata</span>, size=<span class="fl">4</span>)
</pre></div>
<p><img src="BioNERO_files/figure-html/unnamed-chunk-12-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb17"><pre class="downlit">

<span class="co"># Interactive PCA</span>
<span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span>(<span class="kw">finalexp</span>, <span class="kw">metadata</span>, size=<span class="fl">4</span>, interactive=<span class="fl">TRUE</span>)
</pre></div>
<div id="plot_id235543037-container" class="ggvis-output-container">
<div id="plot_id235543037" class="ggvis-output"></div>
<div class="plot-gear-icon">
<nav class="ggvis-control"><a class="ggvis-dropdown-toggle" title="Controls" onclick="return false;"></a>
<ul class="ggvis-dropdown">
<li>
Renderer: 
<a id="plot_id235543037_renderer_svg" class="ggvis-renderer-button" onclick="return false;" data-plot-id="plot_id235543037" data-renderer="svg">SVG</a>
 | 
<a id="plot_id235543037_renderer_canvas" class="ggvis-renderer-button" onclick="return false;" data-plot-id="plot_id235543037" data-renderer="canvas">Canvas</a>
</li>
<li>
<a id="plot_id235543037_download" class="ggvis-download" data-plot-id="plot_id235543037">Download</a>
</li>
</ul></nav>
</div>
</div>
<script type="text/javascript">
var plot_id235543037_spec = {
  "data": [
    {
      "name": "pca_df0",
      "format": {
        "type": "csv",
        "parse": {
          "PC1": "number",
          "PC2": "number",
          "reactive_234084472": "number"
        }
      },
      "values": "\"Trait\",\"SampleID\",\"PC1\",\"PC2\",\"reactive_234084472\"\n\"root\",\"SAMN00009684\",20.0459567222511,-7.86597847110737,50\n\"root\",\"SAMN00009685\",12.076178785311,-8.11525283962293,50\n\"root\",\"SAMN00009686\",-14.7398921779655,1.12479168828691,50\n\"root\",\"SAMN00009687\",12.2121425867538,-3.6616780030144,50\n\"root\",\"SAMN00009688\",-42.4034348941709,9.91790202723798,50\n\"root\",\"SAMN00009689\",-22.5505974429384,-0.788118113640154,50\n\"root\",\"SAMN00009690\",-27.8631206464717,3.59059256721957,50\n\"shoot\",\"SAMN00009691\",-21.3390422735365,-12.313480351835,50\n\"flower\",\"SAMN00009692\",20.2955949384472,-52.0303631436129,50\n\"pod\",\"SAMN00009693\",-26.1724002060486,5.85520166244286,50\n\"leaves\",\"SAMN00009694\",21.9093171413623,-37.782249865588,50\n\"nodule\",\"SAMN00009695\",40.1854132322098,-23.4014229775064,50\n\"root\",\"SAMN00009696\",-55.1145227752498,25.0495414233273,50\n\"root\",\"SAMN00009697\",-24.9558428353804,-12.5500285206913,50\n\"leaves\",\"SAMN02197688\",-9.50821026471291,-16.2200657187537,50\n\"flower\",\"SAMN02197689\",16.8550172233987,-40.7405383661524,50\n\"pod\",\"SAMN02197690\",-33.4892804235877,2.8038034352777,50\n\"pod\",\"SAMN02197691\",-1.23712511267854,-9.96606140096005,50\n\"pod\",\"SAMN02197692\",-21.1084857076576,-14.3107711973964,50\n\"seed\",\"SAMN02197693\",0.0984069658255492,-13.6101682777036,50\n\"seed\",\"SAMN02197696\",-18.901438180934,18.0175268224894,50\n\"seed\",\"SAMN02197697\",-5.29034286091892,32.609414378169,50\n\"seed\",\"SAMN02197698\",15.89175961365,45.0049515021715,50\n\"seed\",\"SAMN02197699\",33.3656271524577,45.9018804988883,50\n\"seed\",\"SAMN02197700\",45.4924251177461,39.9938175327204,50\n\"root\",\"SAMN02197701\",49.7047628538357,38.273080797394,50\n\"nodule\",\"SAMN02197702\",36.541133469003,-14.7863270880407,50"
    },
    {
      "name": "scale/fill",
      "format": {
        "type": "csv",
        "parse": {}
      },
      "values": "\"domain\"\n\"root\"\n\"shoot\"\n\"flower\"\n\"pod\"\n\"leaves\"\n\"nodule\"\n\"seed\""
    },
    {
      "name": "scale/x",
      "format": {
        "type": "csv",
        "parse": {
          "domain": "number"
        }
      },
      "values": "\"domain\"\n-60.3554870567041\n54.94572713529"
    },
    {
      "name": "scale/y",
      "format": {
        "type": "csv",
        "parse": {
          "domain": "number"
        }
      },
      "values": "\"domain\"\n-56.9269753257379\n50.7984926810133"
    }
  ],
  "scales": [
    {
      "name": "fill",
      "type": "ordinal",
      "domain": {
        "data": "scale/fill",
        "field": "data.domain"
      },
      "points": true,
      "sort": false,
      "range": "category10"
    },
    {
      "name": "x",
      "domain": {
        "data": "scale/x",
        "field": "data.domain"
      },
      "zero": false,
      "nice": false,
      "clamp": false,
      "range": "width"
    },
    {
      "name": "y",
      "domain": {
        "data": "scale/y",
        "field": "data.domain"
      },
      "zero": false,
      "nice": false,
      "clamp": false,
      "range": "height"
    }
  ],
  "marks": [
    {
      "type": "symbol",
      "properties": {
        "update": {
          "fill": {
            "scale": "fill",
            "field": "data.Trait"
          },
          "x": {
            "scale": "x",
            "field": "data.PC1"
          },
          "y": {
            "scale": "y",
            "field": "data.PC2"
          },
          "size": {
            "field": "data.reactive_234084472"
          }
        },
        "ggvis": {
          "data": {
            "value": "pca_df0"
          }
        }
      },
      "from": {
        "data": "pca_df0"
      },
      "key": "data.SampleID"
    }
  ],
  "legends": [
    {
      "orient": "right",
      "fill": "fill",
      "title": "Trait"
    }
  ],
  "axes": [
    {
      "type": "x",
      "scale": "x",
      "orient": "bottom",
      "layer": "back",
      "grid": true,
      "title": "PC1"
    },
    {
      "type": "y",
      "scale": "y",
      "orient": "left",
      "layer": "back",
      "grid": true,
      "title": "PC2"
    }
  ],
  "padding": null,
  "ggvis_opts": {
    "keep_aspect": false,
    "resizable": true,
    "padding": {},
    "duration": 250,
    "renderer": "svg",
    "hover_duration": 0,
    "width": 576,
    "height": 480
  },
  "handlers": null
};
ggvis.getPlot("plot_id235543037").parseSpec(plot_id235543037_spec);
</script><p>Other functions exist for exploratory purposes, such as <code><a href="../reference/plot_expression_profile.html">plot_expression_profile()</a></code> and <code><a href="../reference/plot_ngenes_per_module.html">plot_ngenes_per_module()</a></code>, but they were made to be used after gene coexpression network reconstruction or for a set of genes of interest.</p>
</div>
<div id="gene-coexpression-network-reconstruction" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-coexpression-network-reconstruction" class="anchor"></a>3 Gene coexpression network reconstruction</h2>
<p>Now that we have our filtered and normalized expression data, we can reconstruct a gene coexpression network (GCN). First of all, we need to identify the most suitable <span class="math inline">\(\beta\)</span> power that makes the network satisfy the scale-free topology. We do that with the function <code>SFT_fit</code>. Correlation values are raised to a power <span class="math inline">\(\beta\)</span> in order to amplify the distances between correlations and, hence, to make the module detection algorithm more powerful. The higher the value of <span class="math inline">\(\beta\)</span>, the closer to the scale-free topology the network is. However, a very high <span class="math inline">\(\beta\)</span> power reduces mean connectivity, which is not desired. To solve this trade-off, we usually pick the lowest <span class="math inline">\(\beta\)</span> power above a certain threshold (by default in <code>SFT_fit</code>, 0.8). This avoids reducing dramatically the mean connectivity and makes the network topology close to the scale-free topology.</p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="kw">power</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SFT_fit.html">SFT_fit</a></span>(<span class="kw">finalexp</span>, cor_method=<span class="st">"pearson"</span>)
<span class="co">##    Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.</span>
<span class="co">## 1      1  0.77300  6.8900          0.899  2230.0    2270.0   2590</span>
<span class="co">## 2      2  0.53900  2.0700          0.891  1360.0    1380.0   1820</span>
<span class="co">## 3      3  0.22300  0.8730          0.852   885.0     892.0   1350</span>
<span class="co">## 4      4  0.08310  0.4230          0.879   610.0     605.0   1050</span>
<span class="co">## 5      5  0.00248  0.0571          0.868   440.0     430.0    852</span>
<span class="co">## 6      6  0.04960 -0.2210          0.838   330.0     316.0    707</span>
<span class="co">## 7      7  0.21100 -0.4370          0.845   255.0     238.0    605</span>
<span class="co">## 8      8  0.40900 -0.6410          0.861   202.0     182.0    531</span>
<span class="co">## 9      9  0.59300 -0.8240          0.888   164.0     141.0    472</span>
<span class="co">## 10    10  0.68900 -0.9520          0.908   135.0     112.0    426</span>
<span class="co">## 11    11  0.75400 -1.0600          0.923   113.0      90.6    389</span>
<span class="co">## 12    12  0.78400 -1.1500          0.924    95.8      74.7    357</span>
<span class="co">## 13    13  0.81000 -1.2100          0.931    82.1      62.8    330</span>
<span class="co">## 14    14  0.82500 -1.2700          0.931    71.1      52.9    306</span>
<span class="co">## 15    15  0.81600 -1.3400          0.913    62.0      44.6    285</span>
<span class="co">## 16    16  0.84500 -1.3800          0.929    54.6      37.9    267</span>
<span class="co">## 17    17  0.84800 -1.4300          0.921    48.3      32.3    250</span>
<span class="co">## 18    18  0.85100 -1.4700          0.921    43.0      27.7    235</span>
<span class="co">## 19    19  0.86300 -1.4800          0.927    38.5      23.8    222</span>
<span class="co">## 20    20  0.87500 -1.4900          0.936    34.7      20.5    210</span>
<span class="kw">power</span>
<span class="co">## [1] 13</span>
</pre></div>
<p>As we can see, the optimal power is 13. If the users want to visually inspect the simulation of different <span class="math inline">\(\beta\)</span> values, the function <code>SFT_fit</code> automatically saves a PDF figure containing this information.</p>
<p>Now, we can use the power calculated by <code>SFT_fit</code> to reconstruct the GCN. The function <code>exp2net</code> reconstruct a GCN and outputs a list of 5 elements, each of which can be used by other functions in the analysis.</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="kw">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exp2net.html">exp2net</a></span>(<span class="kw">finalexp</span>, net_type=<span class="st">"signed"</span>, SFTpower=<span class="kw">power</span>, cor_method=<span class="st">"pearson"</span>)
<span class="co">## [1] "Calculating adjacency matrix..."</span>
<span class="co">## [1] "Removing diagonals..."</span>
<span class="co">## [1] "Calculating topological overlap matrix (TOM)..."</span>
<span class="co">## ..connectivity..</span>
<span class="co">## ..matrix multiplication (system BLAS)..</span>
<span class="co">## ..normalization..</span>
<span class="co">## ..done.</span>
<span class="co">## [1] "Detecting coexpression modules..."</span>
<span class="co">## [1] "Calculating module eigengenes (MEs)..."</span>
<span class="co">## [1] "Merging similar modules..."</span>
<span class="co">##  mergeCloseModules: Merging modules whose distance is less than 0.2</span>
<span class="co">##    multiSetMEs: Calculating module MEs.</span>
<span class="co">##      Working on set 1 ...</span>
<span class="co">##      moduleEigengenes: Calculating 35 module eigengenes in given set.</span>
<span class="co">##    multiSetMEs: Calculating module MEs.</span>
<span class="co">##      Working on set 1 ...</span>
<span class="co">##      moduleEigengenes: Calculating 19 module eigengenes in given set.</span>
<span class="co">##    multiSetMEs: Calculating module MEs.</span>
<span class="co">##      Working on set 1 ...</span>
<span class="co">##      moduleEigengenes: Calculating 17 module eigengenes in given set.</span>
<span class="co">##    Calculating new MEs...</span>
<span class="co">##    multiSetMEs: Calculating module MEs.</span>
<span class="co">##      Working on set 1 ...</span>
<span class="co">##      moduleEigengenes: Calculating 17 module eigengenes in given set.</span>
<span class="co">## [1] "Calculating intramodular connectivity..."</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">net</span>)
<span class="co">## [1] "adjacency_matrix"   "MEs"                "genes_and_modules" </span>
<span class="co">## [4] "kIN"                "moduleColors"       "correlation_matrix"</span>
</pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Fabricio Almeida-Silva.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
