<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ BioNERO</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioNERO</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/BioNERO.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/almeidasilvaf/BioNERO/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="PC_correction.html">PC_correction()</a></code> </p>
        </td>
        <td><p>Apply Principal Component (PC)-based correction for confounding artifacts in coexpression networks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="SFT_fit.html">SFT_fit()</a></code> </p>
        </td>
        <td><p>Pick power to fit network to a scale-free topology</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ZKfiltering.html">ZKfiltering()</a></code> </p>
        </td>
        <td><p>Filter outlying samples based on standardized connectivity (Zk) method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cel2RMA.html">cel2RMA()</a></code> </p>
        </td>
        <td><p>Create an RMA-normalized expression data frame from .cel files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="consensus_modules.html">consensus_modules()</a></code> </p>
        </td>
        <td><p>Identify consensus modules across multiple data sets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="consensus_set_relationship.html">consensus_set_relationship()</a></code> </p>
        </td>
        <td><p>Relate consensus modules to set-specific modules</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="consensusmodules_sample_cor.html">consensusmodules_sample_cor()</a></code> </p>
        </td>
        <td><p>Correlate set-specific modules and consensus modules to sample information</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="detect_communities.html">detect_communities()</a></code> </p>
        </td>
        <td><p>Detect communities in a network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dfs2one.html">dfs2one()</a></code> </p>
        </td>
        <td><p>Combine multiple expression tables (.tsv) into a single data frame</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="enrichment_analysis.html">enrichment_analysis()</a></code> </p>
        </td>
        <td><p>Perform enrichment analysis for a set of genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="exp2net.html">exp2net()</a></code> </p>
        </td>
        <td><p>Reconstruct gene coexpression network from gene expression data frame</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="exp_preprocess.html">exp_preprocess()</a></code> </p>
        </td>
        <td><p>Preprocess expression data for network reconstruction</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_by_variance.html">filter_by_variance()</a></code> </p>
        </td>
        <td><p>Filter expression data frame to keep only the most varying genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_edges.html">filter_edges()</a></code> </p>
        </td>
        <td><p>Filter edge list to remove spurious correlations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gene_significance.html">gene_significance()</a></code> </p>
        </td>
        <td><p>Calculate gene significance for a given group of genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_HK.html">get_HK()</a></code> </p>
        </td>
        <td><p>Get a vector object with housekeeping genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_edge_list.html">get_edge_list()</a></code> </p>
        </td>
        <td><p>Get edge list from an adjacency matrix for a group of genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_homologs.html">get_homologs()</a></code> </p>
        </td>
        <td><p>Get homologous genes of given gene or gene set</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_hubs.html">get_hubs()</a></code> </p>
        </td>
        <td><p>Get hub genes in a given module or in all modules at once</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_neighbors.html">get_neighbors()</a></code> </p>
        </td>
        <td><p>Get 1st-order neighbors of a given gene or group of genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="igraph2ggnetwork.html">igraph2ggnetwork()</a></code> </p>
        </td>
        <td><p>Create a ggnetwork data frame from an igraph object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is_duplicated.html">is_duplicated()</a></code> </p>
        </td>
        <td><p>Boolean expression to check if gene or gene set is duplicated in the genome or not</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="is_singleton.html">is_singleton()</a></code> </p>
        </td>
        <td><p>Boolean expression to check if gene or gene set is singleton or not</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="module_preservation.html">module_preservation()</a></code> </p>
        </td>
        <td><p>Calculate module preservation between two expression data sets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="module_stability.html">module_stability()</a></code> </p>
        </td>
        <td><p>Perform module stability analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="module_trait_cor.html">module_trait_cor()</a></code> </p>
        </td>
        <td><p>Correlate module eigengenes to trait</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="net_stats.html">net_stats()</a></code> </p>
        </td>
        <td><p>Calculate network statistics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_PCA.html">plot_PCA()</a></code> </p>
        </td>
        <td><p>Plot Principal Component Analysis (PCA) of samples</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_expression_profile.html">plot_expression_profile()</a></code> </p>
        </td>
        <td><p>Plot expression profile of given genes across samples</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_gcn.html">plot_gcn()</a></code> </p>
        </td>
        <td><p>Plot gene coexpression network from edge list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_grn.html">plot_grn()</a></code> </p>
        </td>
        <td><p>Plot gene regulatory network from edge list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_heatmap.html">plot_heatmap()</a></code> </p>
        </td>
        <td><p>Plot heatmap of hierarchically clustered sample correlations or gene expression</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_ngenes_per_module.html">plot_ngenes_per_module()</a></code> </p>
        </td>
        <td><p>Plot number of genes per module</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_ppi.html">plot_ppi()</a></code> </p>
        </td>
        <td><p>Plot protein-protein interaction network from edge list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="probe2gene.html">probe2gene()</a></code> </p>
        </td>
        <td><p>Collapse probe-level expression matrix (or data frame) to gene-level</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="q_normalize.html">q_normalize()</a></code> </p>
        </td>
        <td><p>Quantile normalize the expression data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="remove_na.html">remove_na()</a></code> </p>
        </td>
        <td><p>Remove missing values in a gene expression data frame</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="remove_nonexp.html">remove_nonexp()</a></code> </p>
        </td>
        <td><p>Remove genes that are not expressed based on a user-defined threshold</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Fabricio Almeida-Silva.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.0.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


