<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Gene coexpression network inference • BioNERO</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Gene coexpression network inference">
<meta property="og:description" content="BioNERO">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioNERO</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette_01_GCN_inference.html">Gene coexpression network inference</a>
    </li>
    <li>
      <a href="../articles/vignette_02_GRN_inference.html">Gene regulatory network inference</a>
    </li>
    <li>
      <a href="../articles/vignette_03_network_comparison.html">Network comparison: consensus modules and module preservation</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/almeidasilvaf/BioNERO/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="vignette_01_GCN_inference_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="vignette_01_GCN_inference_files/d3-4.5.0/d3.min.js"></script><script src="vignette_01_GCN_inference_files/forceNetwork-binding-0.4/forceNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Gene coexpression network inference</h1>
                        <h4 data-toc-skip class="author">Fabricio
Almeida-Silva</h4>
            <address class="author_afil">
      Universidade Estadual do Norte Fluminense Darcy Ribeiro, RJ,
Brazil<br><h4 data-toc-skip class="author">Thiago
Motta Venancio</h4>
            <address class="author_afil">
      Universidade Estadual do Norte Fluminense Darcy Ribeiro, RJ,
Brazil<br><small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/BioNERO/blob/HEAD/vignettes/vignette_01_GCN_inference.Rmd" class="external-link"><code>vignettes/vignette_01_GCN_inference.Rmd</code></a></small>
      <div class="hidden name"><code>vignette_01_GCN_inference.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>To date, several packages have been developed to infer gene
coexpression networks from expression data, such as WGCNA <span class="citation">(Langfelder and Horvath 2008)</span>, CEMiTool <span class="citation">(Russo et al. 2018)</span> and petal <span class="citation">(Petereit et al. 2016)</span>. However, network
inference and analysis is a non-trivial task that requires solid
statistical background, especially for data preprocessing and proper
interpretation of results. Because of that, inexperienced researchers
often struggle to choose the most suitable algorithms for their
projects. Besides, different packages are required for each step of a
standard network analysis, and their distinct syntaxes can hinder
interoperability between packages, particularly for non-advanced R
users. Here, we have developed an all-in-one R package that uses
state-of-the-art algorithms to facilitate the workflow of biological
network analysis, from data acquisition to analysis and interpretation.
This will likely accelerate network analysis pipelines and advance
systems biology research.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">'BiocManager'</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BioNERO"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load package after installation</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/BioNERO" class="external-link">BioNERO</a></span><span class="op">)</span>
<span class="co">## </span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co"># for reproducibility</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-loading-and-preprocessing">Data loading and preprocessing<a class="anchor" aria-label="anchor" href="#data-loading-and-preprocessing"></a>
</h2>
<p>For this tutorial, we will use maize (<em>Zea mays</em>) gene
expression data normalized in TPM. The data were obtained from <span class="citation">Shin et al. (2020)</span> and were filtered for package
size issues. For more information on the data set, see
<code><a href="../reference/zma.se.html">?zma.se</a></code>. The data set is stored as a SummarizedExperiment
object.<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></p>
<p>The input expression data in <code>BioNERO</code> can be both a
SummarizedExperiment object or a gene expression matrix or data frame
with genes in rows and samples in columns. However, we strongly
recommend using SummarizedExperiment objects for easier interoperability
with other Bioconductor packages.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">zma.se</span><span class="op">)</span>

<span class="co"># Take a quick look at the data</span>
<span class="va">zma.se</span>
<span class="co">## class: SummarizedExperiment </span>
<span class="co">## dim: 10802 28 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(10802): ZeamMp030 ZeamMp044 ... Zm00001d054106 Zm00001d054107</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(28): SRX339756 SRX339757 ... SRX2792103 SRX2792104</span>
<span class="co">## colData names(1): Tissue</span>
<span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">zma.se</span><span class="op">)</span>
<span class="co">## DataFrame with 28 rows and 1 column</span>
<span class="co">##                    Tissue</span>
<span class="co">##               &lt;character&gt;</span>
<span class="co">## SRX339756       endosperm</span>
<span class="co">## SRX339757       endosperm</span>
<span class="co">## SRX339758       endosperm</span>
<span class="co">## SRX339762       endosperm</span>
<span class="co">## SRX339763       endosperm</span>
<span class="co">## ...                   ...</span>
<span class="co">## SRX2792107 whole_seedling</span>
<span class="co">## SRX2792108 whole_seedling</span>
<span class="co">## SRX2792102 whole_seedling</span>
<span class="co">## SRX2792103 whole_seedling</span>
<span class="co">## SRX2792104 whole_seedling</span></code></pre></div>
<div class="section level3">
<h3 id="step-by-step-data-preprocessing">Step-by-step data preprocessing<a class="anchor" aria-label="anchor" href="#step-by-step-data-preprocessing"></a>
</h3>
<p>This section is suitable for users who want to have more control of
their data analysis, as they can inspect the data set after each
preprocessing step and analyze how different options to the arguments
would affect the expression data. If you want a quick start, you can
skip to the next section (<em>Automatic, one-step data
preprocessing</em>).</p>
<p><strong>Step 1:</strong> Replacing missing values. By default,
<code><a href="../reference/replace_na.html">replace_na()</a></code> will replace NAs with 0. Users can also
replace NAs with the mean of each row (generally not advisable, but it
can be useful in very specific cases).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/replace_na.html">replace_na</a></span><span class="op">(</span><span class="va">zma.se</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">zma.se</span><span class="op">)</span><span class="op">)</span>
<span class="co">## [1] 0</span></code></pre></div>
<p><strong>Step 2:</strong> Removing non-expressed genes. Here, for
faster network reconstruction, we will remove every gene whose median
value is below 10. The function’s default for <code>min_exp</code> is 1.
For other options, see <code><a href="../reference/remove_nonexp.html">?remove_nonexp</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_nonexp.html">remove_nonexp</a></span><span class="op">(</span><span class="va">exp_filt</span>, method<span class="op">=</span><span class="st">"median"</span>, min_exp <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exp_filt</span><span class="op">)</span>
<span class="co">## [1] 8529   28</span></code></pre></div>
<p><strong>Step 3 (optional):</strong> Filtering genes by variance. It
is reasonable to remove genes whose expression values do not vary much
across samples, since we often want to find genes that are more or less
expressed in particular conditions. Here, we will keep only the top 2000
most variable genes. Users can also filter by percentile (e.g., the top
10% most variable genes).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_by_variance.html">filter_by_variance</a></span><span class="op">(</span><span class="va">exp_filt</span>, n<span class="op">=</span><span class="fl">2000</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exp_filt</span><span class="op">)</span>
<span class="co">## [1] 2000   28</span></code></pre></div>
<p><strong>Step 4:</strong> Removing outlying samples. There are several
methods to remove outliers. We have implemented the Z.K (standardized
connectivity) method <span class="citation">(Oldham, Langfelder, and
Horvath 2012)</span> in <code><a href="../reference/ZKfiltering.html">ZKfiltering()</a></code>, which is a
network-based approach to remove outliers. This method has proven to be
more suitable for network analysis, since it can remove outliers that
other methods (such as hierarchical clustering) cannot identify. By
default, BioNERO considers all samples with ZK &lt; 2 as outliers, but
this parameter is flexible if users want to change it.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ZKfiltering.html">ZKfiltering</a></span><span class="op">(</span><span class="va">exp_filt</span>, cor_method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span>
<span class="co">## Number of removed samples: 1</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exp_filt</span><span class="op">)</span>
<span class="co">## [1] 2000   27</span></code></pre></div>
<p><strong>Step 5:</strong> Adjusting for confounding artifacts. This is
an important step to avoid spurious correlations resulting from
confounders. The method was described by <span class="citation">Parsana
et al. (2019)</span>, who developed a principal component (PC)-based
correction for confounders. After correction, the expression data are
quantile normalized, so every gene follows an approximate normal
distribution.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PC_correction.html">PC_correction</a></span><span class="op">(</span><span class="va">exp_filt</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="automatic-one-step-data-preprocessing">Automatic, one-step data preprocessing<a class="anchor" aria-label="anchor" href="#automatic-one-step-data-preprocessing"></a>
</h3>
<p>Alternatively, users can preprocess their data with a single
function. The function <code><a href="../reference/exp_preprocess.html">exp_preprocess()</a></code> is a wrapper for
the functions <code><a href="../reference/replace_na.html">replace_na()</a></code>, <code><a href="../reference/remove_nonexp.html">remove_nonexp()</a></code>,
<code><a href="../reference/filter_by_variance.html">filter_by_variance()</a></code>, <code><a href="../reference/ZKfiltering.html">ZKfiltering()</a></code> and
<code><a href="../reference/PC_correction.html">PC_correction()</a></code>. The arguments passed to
<code><a href="../reference/exp_preprocess.html">exp_preprocess()</a></code> will be used by each of these functions to
generate a filtered expression data frame in a single step.<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">final_exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exp_preprocess.html">exp_preprocess</a></span><span class="op">(</span><span class="va">zma.se</span>, min_exp <span class="op">=</span> <span class="fl">10</span>, variance_filter <span class="op">=</span> <span class="cn">TRUE</span>, n<span class="op">=</span><span class="fl">2000</span><span class="op">)</span>
<span class="co">## Number of removed samples: 1</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">exp_filt</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">final_exp</span><span class="op">)</span>
<span class="co">## [1] TRUE TRUE</span>

<span class="co"># Take a look at the final data</span>
<span class="va">final_exp</span>
<span class="co">## class: SummarizedExperiment </span>
<span class="co">## dim: 2000 27 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): ''</span>
<span class="co">## rownames(2000): ZeamMp030 ZeamMp092 ... Zm00001d054093 Zm00001d054107</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(27): SRX339756 SRX339757 ... SRX2792103 SRX2792104</span>
<span class="co">## colData names(1): Tissue</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="exploratory-data-analysis">Exploratory data analysis<a class="anchor" aria-label="anchor" href="#exploratory-data-analysis"></a>
</h2>
<p><code>BioNERO</code> includes some functions for easy data
exploration. These functions were created to avoid having to type code
chunks that, although small, will be used many times. The idea here is
to make the user experience with biological network analysis as easy and
simple as possible.</p>
<p><strong>Plotting heatmaps:</strong> the function
<code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code> plots heatmaps of correlations between
samples or gene expression in a single line. Users can use their
preferred RColorBrewer’s palette, hide/show gene names, and
activate/deactivate clustering for rows and/or columns.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Heatmap of sample correlations</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">final_exp</span>, type <span class="op">=</span> <span class="st">"samplecor"</span><span class="op">)</span>

<span class="co"># Heatmap of gene expression</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">final_exp</span>, type<span class="op">=</span><span class="st">"expr"</span><span class="op">)</span></code></pre></div>
<p><strong>Principal component analysis (PCA):</strong> the function
<code><a href="../reference/plot_PCA.html">plot_PCA()</a></code> performs a PCA and plots PC1 vs PC2 (by
default), as well the percentage of variance explained by each PC. Users
can also choose to plot PC1 vs PC3 or PC2 vs PC3.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span><span class="op">(</span><span class="va">final_exp</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/pcaplot-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="gene-coexpression-network-inference">Gene coexpression network inference<a class="anchor" aria-label="anchor" href="#gene-coexpression-network-inference"></a>
</h2>
<p>Now that we have our filtered and normalized expression data, we can
reconstruct a gene coexpression network (GCN) with the WGCNA algorithm
<span class="citation">(Langfelder and Horvath 2008)</span>. First of
all, we need to identify the most suitable <span class="math inline">\(\beta\)</span> power that makes the network
satisfy the scale-free topology. We do that with the function
<code><a href="../reference/SFT_fit.html">SFT_fit()</a></code>. Correlation values are raised to a power <span class="math inline">\(\beta\)</span> to amplify their distances and,
hence, to make the module detection algorithm more powerful. The higher
the value of <span class="math inline">\(\beta\)</span>, the closer to
the scale-free topology the network is. However, a very high <span class="math inline">\(\beta\)</span> power reduces mean connectivity,
which is not desired. To solve this trade-off, we pick the lowest <span class="math inline">\(\beta\)</span> power above a certain threshold (by
default in <code><a href="../reference/SFT_fit.html">SFT_fit()</a></code>, 0.8). This makes the network close to
the scale-free topology without dramatically reducing the mean
connectivity.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sft</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SFT_fit.html">SFT_fit</a></span><span class="op">(</span><span class="va">final_exp</span>, net_type<span class="op">=</span><span class="st">"signed hybrid"</span>, cor_method<span class="op">=</span><span class="st">"pearson"</span><span class="op">)</span>
<span class="co">##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.</span>
<span class="co">## 1      3    0.220 -0.218          0.178   278.0    303.00  598.0</span>
<span class="co">## 2      4    0.416 -0.382          0.272   196.0    199.00  472.0</span>
<span class="co">## 3      5    0.573 -0.468          0.462   145.0    136.00  381.0</span>
<span class="co">## 4      6    0.675 -0.536          0.584   110.0     95.70  312.0</span>
<span class="co">## 5      7    0.748 -0.584          0.676    86.3     70.00  259.0</span>
<span class="co">## 6      8    0.791 -0.653          0.735    68.8     51.90  221.0</span>
<span class="co">## 7      9    0.803 -0.717          0.761    55.8     38.60  191.0</span>
<span class="co">## 8     10    0.815 -0.775          0.790    45.8     29.90  167.0</span>
<span class="co">## 9     11    0.821 -0.828          0.815    38.1     22.90  147.0</span>
<span class="co">## 10    12    0.838 -0.874          0.850    32.0     17.90  130.0</span>
<span class="co">## 11    13    0.847 -0.913          0.876    27.2     14.30  116.0</span>
<span class="co">## 12    14    0.856 -0.943          0.893    23.2     11.80  104.0</span>
<span class="co">## 13    15    0.875 -0.973          0.913    20.0      9.79   93.0</span>
<span class="co">## 14    16    0.892 -0.997          0.937    17.3      8.00   83.9</span>
<span class="co">## 15    17    0.897 -1.020          0.941    15.1      6.75   76.0</span>
<span class="co">## 16    18    0.891 -1.070          0.948    13.3      5.79   69.7</span>
<span class="co">## 17    19    0.888 -1.100          0.950    11.7      4.96   64.2</span>
<span class="co">## 18    20    0.888 -1.130          0.957    10.4      4.27   59.4</span>
<span class="va">sft</span><span class="op">$</span><span class="va">power</span>
<span class="co">## [1] 9</span>
<span class="va">power</span> <span class="op">&lt;-</span> <span class="va">sft</span><span class="op">$</span><span class="va">power</span></code></pre></div>
<p>As we can see, the optimal power is 9. However, we <strong>strongly
recommend</strong> a visual inspection of the simulation of different
<span class="math inline">\(\beta\)</span> powers, as WGCNA can fail to
return the most appropriate <span class="math inline">\(\beta\)</span>
power in some cases.<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a> The function <code><a href="../reference/SFT_fit.html">SFT_fit()</a></code>
automatically saves a ggplot object in the second element of the
resulting list. To visualize it, you simply have to access the plot.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sft</span><span class="op">$</span><span class="va">plot</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/plot_sft-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Now, we can use the power calculated by <code><a href="../reference/SFT_fit.html">SFT_fit()</a></code> to
infer the GCN. The function <code><a href="../reference/exp2gcn.html">exp2gcn()</a></code> infers a GCN and
outputs a list of 7 elements, each of which will be used by other
functions in the analysis pipeline.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exp2gcn.html">exp2gcn</a></span><span class="op">(</span><span class="va">final_exp</span>, net_type<span class="op">=</span><span class="st">"signed hybrid"</span>, 
               SFTpower<span class="op">=</span><span class="va">power</span>, cor_method<span class="op">=</span><span class="st">"pearson"</span><span class="op">)</span>
<span class="co">## ..connectivity..</span>
<span class="co">## ..matrix multiplication (system BLAS)..</span>
<span class="co">## ..normalization..</span>
<span class="co">## ..done.</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span>
<span class="co">## [1] "adjacency_matrix"    "MEs"                 "genes_and_modules"  </span>
<span class="co">## [4] "kIN"                 "moduleColors"        "correlation_matrix" </span>
<span class="co">## [7] "params"              "dendro_plot_objects"</span></code></pre></div>
<p>The function <code><a href="../reference/exp2gcn.html">exp2gcn()</a></code> saves objects in the last element
of the resulting list that can be subsequently used to plot common
figures in GCN papers. The figures are publication-ready and display i.
a dendrogram of genes and modules; ii. heatmap of pairwise correlations
between module eigengenes.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Dendro and colors</span>
<span class="fu"><a href="../reference/plot_dendro_and_colors.html">plot_dendro_and_colors</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/dendro-1.png" width="480" style="display: block; margin: auto;"></p>
<pre><code><span class="co">## NULL</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Eigengene networks</span>
<span class="fu"><a href="../reference/plot_eigengene_network.html">plot_eigengene_network</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/eigengene_net-1.png" width="480" style="display: block; margin: auto;"></p>
<pre><code><span class="co">## NULL</span></code></pre>
<p>Let’s see the number of genes per module.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_ngenes_per_module.html">plot_ngenes_per_module</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/genes_per_module-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="gene-coexpression-network-analysis">Gene coexpression network analysis<a class="anchor" aria-label="anchor" href="#gene-coexpression-network-analysis"></a>
</h2>
<p>Now that we have our coexpression network, we can start exploring
some of its properties.</p>
<div class="section level3">
<h3 id="assessing-module-stability">Assessing module stability<a class="anchor" aria-label="anchor" href="#assessing-module-stability"></a>
</h3>
<p>The function <code><a href="../reference/module_stability.html">module_stability()</a></code> allows users to check if
the identified coexpression modules are stable (i.e., if they can resist
removal of a particular sample). This function will resample the data
set and rerun the module detection algorithm <em>n</em> times (default:
30) and return a PDF figure displaying a gene dendrogram and colors
representing modules identified in each run. By looking at the figure,
you can detect if a particular module is only found in a very small
fraction of the runs, which suggests instability. Here, we will perform
only 5 resampling runs for demonstration purposes.<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/module_stability.html">module_stability</a></span><span class="op">(</span><span class="va">final_exp</span>, <span class="va">net</span>, nRuns<span class="op">=</span><span class="fl">5</span><span class="op">)</span>
<span class="co">##  ...working on run 1 ..</span>
<span class="co">##  ...working on run 2 ..</span>
<span class="co">##  ...working on run 3 ..</span>
<span class="co">##  ...working on run 4 ..</span>
<span class="co">##  ...working on run 5 ..</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/module_stability-1.png" width="480" style="display: block; margin: auto;"></p>
<pre><code><span class="co">## NULL</span></code></pre>
</div>
<div class="section level3">
<h3 id="module-trait-associations">Module-trait associations<a class="anchor" aria-label="anchor" href="#module-trait-associations"></a>
</h3>
<p>The function <code><a href="../reference/module_trait_cor.html">module_trait_cor()</a></code> can be used to calculate
module-trait correlations. This analysis is useful to identify modules
that are positively or negatively correlated with particular traits,
which means that their gene expression levels go up or down in these
conditions. Here, tissues will be considered traits, so we want to
identify groups of genes whose expression levels are inhibited or
induced in particular tissues. Alternatively, one can use continuous
variables (e.g., metabolite content, protein concentration, height) or
discrete variables (e.g., disease index) as traits.<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MEtrait</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/module_trait_cor.html">module_trait_cor</a></span><span class="op">(</span>exp<span class="op">=</span><span class="va">final_exp</span>, MEs<span class="op">=</span><span class="va">net</span><span class="op">$</span><span class="va">MEs</span>, cor_method<span class="op">=</span><span class="st">"pearson"</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/moddtraitcor-1.png" width="480" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">MEtrait</span><span class="op">)</span>
<span class="co">##        ME          trait         cor    pvalue</span>
<span class="co">## 1 MEblack      endosperm  0.12925374 0.5205188</span>
<span class="co">## 2 MEblack         pollen  0.11154977 0.5796226</span>
<span class="co">## 3 MEblack whole_seedling -0.20119355 0.3142702</span>
<span class="co">## 4 MEbrown      endosperm -0.02538113 0.8999982</span>
<span class="co">## 5 MEbrown         pollen  0.06199200 0.7587116</span>
<span class="co">## 6 MEbrown whole_seedling -0.02607732 0.8972693</span></code></pre></div>
<p>The function <code><a href="../reference/module_trait_cor.html">module_trait_cor()</a></code> also allows for plot
customization. For instance:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Transpose the matrix and change palette (RColorBrewer palette)</span>
<span class="va">MEtrait</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/module_trait_cor.html">module_trait_cor</a></span><span class="op">(</span>exp<span class="op">=</span><span class="va">final_exp</span>, MEs<span class="op">=</span><span class="va">net</span><span class="op">$</span><span class="va">MEs</span>, cor_method<span class="op">=</span><span class="st">"pearson"</span>,
                            transpose <span class="op">=</span> <span class="cn">TRUE</span>, palette<span class="op">=</span><span class="st">"PRGn"</span>, cex.text <span class="op">=</span> <span class="fl">0.7</span>,
                            cex.lab.y <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/modtraitcor_transpose-1.png" width="864" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="visualizing-module-expression-profile">Visualizing module expression profile<a class="anchor" aria-label="anchor" href="#visualizing-module-expression-profile"></a>
</h3>
<p>The heatmap above shows that genes in the <em>yellow</em> module are
negatively correlated with endosperm samples. We can visually explore it
with <code><a href="../reference/plot_expression_profile.html">plot_expression_profile()</a></code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_expression_profile.html">plot_expression_profile</a></span><span class="op">(</span>exp<span class="op">=</span><span class="va">final_exp</span>, net<span class="op">=</span><span class="va">net</span>, 
                        plot_module<span class="op">=</span><span class="cn">TRUE</span>, modulename<span class="op">=</span><span class="st">"yellow"</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/exp_profile-1.png" width="384" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="enrichment-analysis">Enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h3>
<p>After identifying modules that are inhibited or enhanced in
particular tissues, users would likely want to find to which biological
processes (e.g., GO biological process) or pathways (e.g., Reactome,
KEGG, MapMan) these genes are related. This can be done with enrichment
analyses, which can uncover terms that are found more than expected by
chance in a module.</p>
<p>The easiest way to accomplish this is to use the function
<code><a href="../reference/module_enrichment.html">module_enrichment()</a></code>, which performs enrichment analysis for
all modules at once. To illustrate it, we will scan coexpression modules
for enriched protein domains using all genes in the network as
background. The Interpro annotation was downloaded from the PLAZA 4.0
Monocots database <span class="citation">(Van Bel et al.
2018)</span>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Enrichment analysis for conserved protein domains (Interpro)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">zma.interpro</span><span class="op">)</span>
<span class="va">interpro_enrichment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/module_enrichment.html">module_enrichment</a></span><span class="op">(</span>net <span class="op">=</span> <span class="va">net</span>, 
                                         background_genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">final_exp</span><span class="op">)</span>,
                                         annotation <span class="op">=</span> <span class="va">zma.interpro</span><span class="op">)</span>
<span class="co">## Enrichment analysis for module black...</span>
<span class="co">## Enrichment analysis for module brown...</span>
<span class="co">## Enrichment analysis for module darkgreen...</span>
<span class="co">## Enrichment analysis for module darkolivegreen...</span>
<span class="co">## Enrichment analysis for module greenyellow...</span>
<span class="co">## Enrichment analysis for module lightyellow...</span>
<span class="co">## Enrichment analysis for module midnightblue...</span>
<span class="co">## Enrichment analysis for module paleturquoise...</span>
<span class="co">## Enrichment analysis for module violet...</span>
<span class="co">## Enrichment analysis for module yellow...</span>

<span class="co"># Print results without geneIDs for better visualization</span>
<span class="va">interpro_enrichment</span><span class="op">[</span>, <span class="op">-</span><span class="fl">6</span><span class="op">]</span>
<span class="co">##                                       TermID genes all         pval</span>
<span class="co">## 809                             Histone-fold    58  60 1.083394e-11</span>
<span class="co">## 802                       Histone H2A/H2B/H3    43  44 2.155952e-09</span>
<span class="co">## 1877     Translation protein SH3-like domain    22  22 8.872064e-06</span>
<span class="co">## 1878 Translation protein, beta-barrel domain    26  27 1.235834e-05</span>
<span class="co">## 1394           Ribosomal protein L2 domain 2    18  18 7.448332e-05</span>
<span class="co">## 186                    Aquaporin transporter     3   5 9.644015e-05</span>
<span class="co">## 187                           Aquaporin-like     3   5 9.644015e-05</span>
<span class="co">## 906                  Major intrinsic protein     3   5 9.644015e-05</span>
<span class="co">## 907  Major intrinsic protein, conserved site     3   5 9.644015e-05</span>
<span class="co">##              padj       Module</span>
<span class="co">## 809  2.178705e-08        black</span>
<span class="co">## 802  2.167810e-06        black</span>
<span class="co">## 1877 5.947240e-03        black</span>
<span class="co">## 1878 6.213155e-03        black</span>
<span class="co">## 1394 2.995719e-02        black</span>
<span class="co">## 186  4.848528e-02 midnightblue</span>
<span class="co">## 187  4.848528e-02 midnightblue</span>
<span class="co">## 906  4.848528e-02 midnightblue</span>
<span class="co">## 907  4.848528e-02 midnightblue</span></code></pre></div>
<p>As we can see, two modules are enriched in genes with particular
protein domains. We could get the same result with the function
<code><a href="../reference/enrichment_analysis.html">enrichment_analysis()</a></code>, which performs enrichment analysis
for a user-defined gene set instead of all modules. <a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a></p>
</div>
<div class="section level3">
<h3 id="hub-gene-identification">Hub gene identification<a class="anchor" aria-label="anchor" href="#hub-gene-identification"></a>
</h3>
<p>Hub genes are often identified using two different metrics:
<strong>module membership (MM)</strong> (i.e., correlation of a gene to
its module eigengene) and <strong>degree</strong> (i.e., sum of
connection weights of a gene to all other genes in the module). Some
researchers consider the top 10% genes with the highest degree as hubs,
while others consider those with MM &gt; 0.8. To avoid false positives,
<code>BioNERO</code>’s algorithm combines both metrics and defines hub
genes as the top 10% genes with highest degree that have MM &gt; 0.8.
Hubs can be identified with the function
<code><a href="../reference/get_hubs_gcn.html">get_hubs_gcn()</a></code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hubs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_hubs_gcn.html">get_hubs_gcn</a></span><span class="op">(</span><span class="va">final_exp</span>, <span class="va">net</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hubs</span><span class="op">)</span>
<span class="co">##             Gene Module  kWithin</span>
<span class="co">## 1 Zm00001d033147  black 188.3864</span>
<span class="co">## 2 Zm00001d049790  black 181.4522</span>
<span class="co">## 3 Zm00001d005649  black 180.7062</span>
<span class="co">## 4 Zm00001d045448  black 180.6744</span>
<span class="co">## 5 Zm00001d008203  black 178.7147</span>
<span class="co">## 6 Zm00001d023340  black 177.7553</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extracting-subgraphs">Extracting subgraphs<a class="anchor" aria-label="anchor" href="#extracting-subgraphs"></a>
</h3>
<p>Subgraph extraction can be particularly useful to visualize specific
modules, and it can be done with the function
<code><a href="../reference/get_edge_list.html">get_edge_list()</a></code>. The function returns the subgraph as an
edge list. Users can also extract an edge list for a particular gene set
instead of a module.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">edges</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_edge_list.html">get_edge_list</a></span><span class="op">(</span><span class="va">net</span>, module<span class="op">=</span><span class="st">"midnightblue"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">edges</span><span class="op">)</span>
<span class="co">##              Gene1          Gene2    Weight</span>
<span class="co">## 45  Zm00001d001857 Zm00001d002384 0.9401886</span>
<span class="co">## 89  Zm00001d001857 Zm00001d002690 0.9675345</span>
<span class="co">## 90  Zm00001d002384 Zm00001d002690 0.9185426</span>
<span class="co">## 133 Zm00001d001857 Zm00001d003962 0.7178340</span>
<span class="co">## 134 Zm00001d002384 Zm00001d003962 0.6534956</span>
<span class="co">## 135 Zm00001d002690 Zm00001d003962 0.6840004</span></code></pre></div>
<p>The function <code><a href="../reference/get_edge_list.html">get_edge_list()</a></code> returns a fully connected
subgraph for the specified module or gene set. However, filtering weak
correlations is desirable and can be accomplished by setting the
argument <code>filter = TRUE</code>, which will remove edges based on
one of optimal scale-free topology fit (default), p-value, Z-score, or
an arbitrary minimum correlation coefficient. <a href="#fn7" class="footnote-ref" id="fnref7"><sup>7</sup></a> For more details
details, check <code>?get_edge_list()</code>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove edges based on optimal scale-free topology fit</span>
<span class="va">edges_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_edge_list.html">get_edge_list</a></span><span class="op">(</span><span class="va">net</span>, module<span class="op">=</span><span class="st">"midnightblue"</span>,
                                filter<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## The correlation threshold that best fits the scale-free topology is 0.7</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/filter_edges-1.png" width="480" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">edges_filtered</span><span class="op">)</span>
<span class="co">## [1] 588   3</span>

<span class="co"># Remove edges based on p-value</span>
<span class="va">edges_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_edge_list.html">get_edge_list</a></span><span class="op">(</span><span class="va">net</span>, module<span class="op">=</span><span class="st">"midnightblue"</span>,
                                filter<span class="op">=</span><span class="cn">TRUE</span>, method<span class="op">=</span><span class="st">"pvalue"</span>,
                                nSamples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">final_exp</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">edges_filtered</span><span class="op">)</span>
<span class="co">## [1] 921   3</span>

<span class="co"># Remove edges based on minimum correlation</span>
<span class="va">edges_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_edge_list.html">get_edge_list</a></span><span class="op">(</span><span class="va">net</span>, module<span class="op">=</span><span class="st">"midnightblue"</span>,
                                filter<span class="op">=</span><span class="cn">TRUE</span>, method<span class="op">=</span><span class="st">"min_cor"</span>,
                                rcutoff <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">edges_filtered</span><span class="op">)</span>
<span class="co">## [1] 588   3</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="network-visualization">Network visualization<a class="anchor" aria-label="anchor" href="#network-visualization"></a>
</h3>
<p>As we now have an edge list for a module, let’s visualize it with the
function <code><a href="../reference/plot_gcn.html">plot_gcn()</a></code>. By default, this function only labels
the top 5 hubs (or less if there are less than 5 hubs). However, this
can be customized according to the user’s preference (see
<code><a href="../reference/plot_gcn.html">?plot_gcn</a></code> for more information).</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_gcn.html">plot_gcn</a></span><span class="op">(</span>edgelist_gcn <span class="op">=</span> <span class="va">edges_filtered</span>, 
         net <span class="op">=</span> <span class="va">net</span>,
         color_by <span class="op">=</span> <span class="st">"module"</span>,
         hubs <span class="op">=</span> <span class="va">hubs</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_01_GCN_inference_files/figure-html/plot_gcn-1.png" width="384" style="display: block; margin: auto;"></p>
<p>Networks can also be visualized interactively by setting
<code>interactive = TRUE</code> in <code>plot_gcn</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_gcn.html">plot_gcn</a></span><span class="op">(</span>edgelist_gcn <span class="op">=</span> <span class="va">edges_filtered</span>, 
         net <span class="op">=</span> <span class="va">net</span>,
         color_by<span class="op">=</span><span class="st">"module"</span>,
         hubs <span class="op">=</span> <span class="va">hubs</span>,
         interactive <span class="op">=</span> <span class="cn">TRUE</span>,
         dim_interactive <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">500</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-4c3cc5ef36d163ce2865" style="width:500px;height:500px;" class="forceNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-4c3cc5ef36d163ce2865">{"x":{"links":{"source":[2,3,0,4,1,11,12,7,5,6,8,0,3,4,1,10,9,2,3,1,7,4,5,0,2,6,25,14,20,2,0,16,7,3,5,30,17,29,6,23,12,1,10,28,33,15,10,16,7,4,5,3,0,1,12,9,20,23,25,15,2,26,8,6,17,27,5,4,24,1,15,10,14,0,23,19,3,12,25,16,6,7,20,8,11,9,13,17,2,4,5,0,2,3,1,6,7,8,5,1,14,0,12,15,6,10,7,16,8,11,4,9,3,17,13,20,23,2,37,5,4,1,3,2,0,6,2,8,3,0,1,4,5,10,9,7,5,14,1,13,16,3,9,11,0,8,12,2,7,10,17,6,4,23,6,7,15,34,14,10,25,33,3,0,5,14,16,20,13,7,6,1,8,22,10,15,19,9,3,2,17,4,12,5,4,1,7,15,0,13,9,14,12,10,6,11,8,2,3,5,6,14,17,12,13,23,26,4,7,0,10,11,21,3,1,15,25,26,12,16,25,10,5,0,13,23,15,20,9,8,17,3,6,1,4,2,3,13,17,5,24,16,9,15,14,2,12,4,7,19,18,1,8,0,6,11,22,35,23,7,20,31,24,10,14,8,9,22,13,15,7,3,19,21,12,1,17,23,5,0,11,6,4,2,20,18,22,32,7,24,31,13,21,14,0,2,1,6,14,21,31,24,36,22,32,7,35,5,8,0,11,1,4,12,6,7,13,3,9,2,10,6,0,8,3,1,2,4,7,9,5,8,11,3,6,0,4,7,9,2,5,10,1,4,6,3,5,1,2,0,3,9,0,7,10,5,12,8,4,21,14,20,18,17,6,2,11,15,1,13,19,2,14,9,3,15,0,23,28,7,20,32,1,25,10,22,33,31,12,6,4,9,4,1,2,6,3,35,27,12,23,25,7,10,20,28,15,0,24,3,14,13,12,15,19,2,1,8,5,21,17,29,7,6,18,16,11,0,2,0,1,3,7,13,16,3,2,5,0,15,12,9,14,10,6,11,4,17,1,8,18,16,0,15,12,7,9,5,3,8,11,1,14,10,13,6,4,2,7,14,0,11,4,5,12,13,6,1,10,9,8,2,3,0,5,16,7,0,10,1,15,8,13,14,3,6,17,9,12,11,4,19,2,16,3,10,5,15,0,7,9,14,12,8,17,2,20,6,4,1,13,11,18,19,0,22,5,14,3,15,13,4,7,10,18,19,20,8,2,6,9,11,12,1,17,16,21,14,11,7,1,6,5,3,13,4,22,2,26,0,23,24,9,15,12,21,31,31,37,22,36,15,3,4,23,10,20,0,7,1,30,25,12,6,14,11,1,0],"target":[5,5,5,5,5,13,13,13,13,13,13,13,13,13,13,13,13,13,8,8,8,8,8,8,8,8,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,34,28,28,28,28,28,28,28,28,28,28,28,28,28,28,28,28,28,28,28,28,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,26,9,9,9,9,9,9,9,9,9,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,42,6,6,6,6,6,6,11,11,11,11,11,11,11,11,11,11,11,18,18,18,18,18,18,18,18,18,18,18,18,18,18,18,18,18,40,40,40,40,40,40,40,40,40,40,23,23,23,23,23,23,23,23,23,23,23,23,23,23,23,23,23,23,23,23,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,38,38,38,38,38,38,38,38,38,38,38,38,38,38,38,38,38,38,27,27,27,27,27,27,27,27,27,27,27,27,27,27,27,27,27,27,27,29,29,29,29,29,29,29,29,29,29,29,29,29,29,29,29,29,29,29,29,41,41,41,41,41,41,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,31,36,36,36,36,36,36,36,36,3,3,3,37,37,37,37,37,37,37,37,37,37,14,14,14,14,14,14,14,14,14,14,14,14,14,14,10,10,10,10,10,10,10,10,10,10,12,12,12,12,12,12,12,12,12,12,12,12,7,7,7,7,7,7,7,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,35,35,35,35,35,35,35,35,35,35,35,35,35,35,35,35,35,35,35,35,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,4,4,4,4,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,15,15,15,15,15,15,15,15,15,15,15,15,15,15,15,1,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,21,21,21,21,21,21,21,21,21,21,21,21,21,21,21,21,21,21,21,21,21,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,24,32,32,32,32,32,32,32,32,32,32,32,32,32,32,32,32,32,32,32,32,43,43,43,43,33,33,33,33,33,33,33,33,33,33,33,33,33,33,33,2,2],"colour":["#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666"]},"nodes":{"name":["Zm00001d034368","Zm00001d042976","Zm00001d051362","Zm00001d028375","Zm00001d038136","Zm00001d001857","Zm00001d011034","Zm00001d033707","Zm00001d002690","Zm00001d007234","Zm00001d031569","Zm00001d011081","Zm00001d033038","Zm00001d002384","Zm00001d029378","Zm00001d042372","Zm00001d016733","Zm00001d040803","Zm00001d014781","Zm00001d038218","Zm00001d043299","Zm00001d043998","Zm00001d034828","Zm00001d015918","Zm00001d047204","Zm00001d008305","Zm00001d006458","Zm00001d019563","Zm00001d005421","Zm00001d021813","Zm00001d037018","Zm00001d023559","Zm00001d048693","Zm00001d049691","Zm00001d003962","Zm00001d036162","Zm00001d025892","Zm00001d028630","Zm00001d019553","Zm00001d036547","Zm00001d015399","Zm00001d022464","Zm00001d008600","Zm00001d048802"],"group":["midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue","midnightblue"],"nodesize":[10.5817658089403,10.2457134175312,10.1012265821005,9.20536757438946,9.09708978527968,9.02890397173426,8.90902323921577,8.8960519819567,8.83398655496156,8.17926469128355,8.07960378246698,7.95166881927189,7.48687403118163,7.46018171904601,7.39303782719813,7.29407548122694,7.20544171499372,6.71003994604486,5.56968360957348,5.2513403679063,5.22106943556366,5.21776630620915,4.99594484674438,4.77413047577239,4.77324261059502,4.39060194558006,4.10510591167904,3.79870217969408,3.74610910181984,3.47086972584609,3.35860606173085,3.31743803885518,3.26312724068693,3.21699033303724,2.70939551612667,2.56173369044579,1.90480419574236,1.58878210422147,1.56723036367367,1.47677712893716,1.38039516028772,1.00125948426605,0.485854283155314,0.364633092930587]},"options":{"NodeID":"name","Group":"group","colourScale":"d3.scaleOrdinal(d3.schemeCategory20);","fontSize":13,"fontFamily":"serif","clickTextSize":32.5,"linkDistance":50,"linkWidth":"function(d) { return Math.sqrt(d.value); }","charge":-30,"opacity":0.8,"zoom":true,"legend":false,"arrows":false,"nodesize":true,"radiusCalculation":" Math.sqrt(d.nodesize)+6","bounded":false,"opacityNoHover":0,"clickAction":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 id="network-statistics">Network statistics<a class="anchor" aria-label="anchor" href="#network-statistics"></a>
</h3>
<p>Finally, the function <code><a href="../reference/net_stats.html">net_stats()</a></code> can be used to
calculate the main network statistics (or properties, or indices),
namely: <em>connectivity</em>, <em>scaled connectivity</em>,
<em>clustering coefficient</em>, <em>maximum adjacency ratio</em>,
<em>density</em>, <em>centralization</em>, <em>heterogeneity</em>,
<em>number of cliques</em>, <em>diameter</em>, <em>betweenness</em>
(optional), and <em>closeness</em> (optional).</p>
<p>Depending on your system capacities and network size, this may take a
very long time. Hence, if you are willing to calculate network
statistics for your data set, grab a cup of coffee, because the waiting
may be long.</p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This vignette was created under the following conditions:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.4 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] BioNERO_1.3.1    BiocStyle_2.22.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] backports_1.4.1             circlize_0.4.14            </span>
<span class="co">##   [3] Hmisc_4.7-0                 systemfonts_1.0.4          </span>
<span class="co">##   [5] plyr_1.8.7                  igraph_1.3.0               </span>
<span class="co">##   [7] splines_4.1.3               GENIE3_1.16.0              </span>
<span class="co">##   [9] BiocParallel_1.28.3         ggnetwork_0.5.10           </span>
<span class="co">##  [11] GenomeInfoDb_1.30.1         ggplot2_3.3.5              </span>
<span class="co">##  [13] sva_3.42.0                  digest_0.6.29              </span>
<span class="co">##  [15] foreach_1.5.2               htmltools_0.5.2            </span>
<span class="co">##  [17] GO.db_3.14.0                fansi_1.0.3                </span>
<span class="co">##  [19] magrittr_2.0.3              checkmate_2.0.0            </span>
<span class="co">##  [21] memoise_2.0.1               cluster_2.1.3              </span>
<span class="co">##  [23] doParallel_1.0.17           limma_3.50.3               </span>
<span class="co">##  [25] ComplexHeatmap_2.10.0       fastcluster_1.2.3          </span>
<span class="co">##  [27] Biostrings_2.62.0           annotate_1.72.0            </span>
<span class="co">##  [29] matrixStats_0.62.0          pkgdown_2.0.2              </span>
<span class="co">##  [31] jpeg_0.1-9                  colorspace_2.0-3           </span>
<span class="co">##  [33] ggrepel_0.9.1               blob_1.2.3                 </span>
<span class="co">##  [35] textshaping_0.3.6           xfun_0.30                  </span>
<span class="co">##  [37] dplyr_1.0.8                 crayon_1.5.1               </span>
<span class="co">##  [39] RCurl_1.98-1.6              jsonlite_1.8.0             </span>
<span class="co">##  [41] genefilter_1.76.0           impute_1.68.0              </span>
<span class="co">##  [43] survival_3.3-1              iterators_1.0.14           </span>
<span class="co">##  [45] glue_1.6.2                  gtable_0.3.0               </span>
<span class="co">##  [47] zlibbioc_1.40.0             XVector_0.34.0             </span>
<span class="co">##  [49] GetoptLong_1.0.5            DelayedArray_0.20.0        </span>
<span class="co">##  [51] car_3.0-12                  shape_1.4.6                </span>
<span class="co">##  [53] BiocGenerics_0.40.0         abind_1.4-5                </span>
<span class="co">##  [55] scales_1.2.0                edgeR_3.36.0               </span>
<span class="co">##  [57] DBI_1.1.2                   rstatix_0.7.0              </span>
<span class="co">##  [59] Rcpp_1.0.8.3                xtable_1.8-4               </span>
<span class="co">##  [61] htmlTable_2.4.0             clue_0.3-60                </span>
<span class="co">##  [63] foreign_0.8-82              bit_4.0.4                  </span>
<span class="co">##  [65] preprocessCore_1.56.0       Formula_1.2-4              </span>
<span class="co">##  [67] stats4_4.1.3                htmlwidgets_1.5.4          </span>
<span class="co">##  [69] httr_1.4.2                  RColorBrewer_1.1-3         </span>
<span class="co">##  [71] ellipsis_0.3.2              farver_2.1.0               </span>
<span class="co">##  [73] pkgconfig_2.0.3             XML_3.99-0.9               </span>
<span class="co">##  [75] nnet_7.3-17                 sass_0.4.1                 </span>
<span class="co">##  [77] locfit_1.5-9.5              utf8_1.2.2                 </span>
<span class="co">##  [79] dynamicTreeCut_1.63-1       labeling_0.4.2             </span>
<span class="co">##  [81] reshape2_1.4.4              tidyselect_1.1.2           </span>
<span class="co">##  [83] rlang_1.0.2                 AnnotationDbi_1.56.2       </span>
<span class="co">##  [85] munsell_0.5.0               tools_4.1.3                </span>
<span class="co">##  [87] cachem_1.0.6                cli_3.2.0                  </span>
<span class="co">##  [89] generics_0.1.2              RSQLite_2.2.12             </span>
<span class="co">##  [91] statnet.common_4.5.0        broom_0.8.0                </span>
<span class="co">##  [93] evaluate_0.15               stringr_1.4.0              </span>
<span class="co">##  [95] fastmap_1.1.0               yaml_2.3.5                 </span>
<span class="co">##  [97] ragg_1.2.2                  RhpcBLASctl_0.21-247.1     </span>
<span class="co">##  [99] knitr_1.38                  bit64_4.0.5                </span>
<span class="co">## [101] fs_1.5.2                    purrr_0.3.4                </span>
<span class="co">## [103] KEGGREST_1.34.0             nlme_3.1-157               </span>
<span class="co">## [105] compiler_4.1.3              rstudioapi_0.13            </span>
<span class="co">## [107] png_0.1-7                   ggsignif_0.6.3             </span>
<span class="co">## [109] minet_3.52.0                tibble_3.1.6               </span>
<span class="co">## [111] statmod_1.4.36              geneplotter_1.72.0         </span>
<span class="co">## [113] bslib_0.3.1                 stringi_1.7.6              </span>
<span class="co">## [115] highr_0.9                   desc_1.4.1                 </span>
<span class="co">## [117] lattice_0.20-45             Matrix_1.4-1               </span>
<span class="co">## [119] vctrs_0.4.1                 networkD3_0.4              </span>
<span class="co">## [121] pillar_1.7.0                lifecycle_1.0.1            </span>
<span class="co">## [123] BiocManager_1.30.16         jquerylib_0.1.4            </span>
<span class="co">## [125] GlobalOptions_0.1.2         cowplot_1.1.1              </span>
<span class="co">## [127] data.table_1.14.2           bitops_1.0-7               </span>
<span class="co">## [129] GenomicRanges_1.46.1        R6_2.5.1                   </span>
<span class="co">## [131] latticeExtra_0.6-29         bookdown_0.26              </span>
<span class="co">## [133] network_1.17.1              gridExtra_2.3              </span>
<span class="co">## [135] IRanges_2.28.0              codetools_0.2-18           </span>
<span class="co">## [137] SummarizedExperiment_1.24.0 DESeq2_1.34.0              </span>
<span class="co">## [139] rprojroot_2.0.3             rjson_0.2.21               </span>
<span class="co">## [141] S4Vectors_0.32.4            GenomeInfoDbData_1.2.7     </span>
<span class="co">## [143] intergraph_2.0-2            mgcv_1.8-40                </span>
<span class="co">## [145] parallel_4.1.3              grid_4.1.3                 </span>
<span class="co">## [147] rpart_4.1.16                tidyr_1.2.0                </span>
<span class="co">## [149] NetRep_1.2.4                coda_0.19-4                </span>
<span class="co">## [151] rmarkdown_2.13              carData_3.0-5              </span>
<span class="co">## [153] MatrixGenerics_1.6.0        ggpubr_0.4.0               </span>
<span class="co">## [155] ggnewscale_0.4.7            Biobase_2.54.0             </span>
<span class="co">## [157] WGCNA_1.70-3                base64enc_0.1-3</span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Langfelder2008" class="csl-entry">
Langfelder, Peter, and Steve Horvath. 2008. <span>“<span class="nocase">WGCNA: an R package for weighted correlation network
analysis</span>.”</span> <em>BMC Bioinformatics</em> 9 (1): 559. <a href="https://doi.org/10.1186/1471-2105-9-559" class="external-link">https://doi.org/10.1186/1471-2105-9-559</a>.
</div>
<div id="ref-Love2014" class="csl-entry">
Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. <span>“<span class="nocase">Moderated estimation of fold change and dispersion for
RNA-seq data with DESeq2</span>.”</span> <em>Genome Biology</em> 15
(12): 1–21. <a href="https://doi.org/10.1186/s13059-014-0550-8" class="external-link">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div id="ref-Oldham2012" class="csl-entry">
Oldham, Michael C., Peter Langfelder, and Steve Horvath. 2012.
<span>“<span class="nocase">Network methods for describing sample
relationships in genomic datasets: application to Huntington’s
disease</span>.”</span> <em>BMC Systems Biology</em> 6 (1): 1. <a href="https://doi.org/10.1186/1752-0509-6-63" class="external-link">https://doi.org/10.1186/1752-0509-6-63</a>.
</div>
<div id="ref-Parsana2019" class="csl-entry">
Parsana, Princy, Claire Ruberman, Andrew E. Jaffe, Michael C. Schatz,
Alexis Battle, and Jeffrey T. Leek. 2019. <span>“<span class="nocase">Addressing confounding artifacts in reconstruction of
gene co-expression networks</span>.”</span> <em>Genome Biology</em> 20
(1): 94. <a href="https://doi.org/10.1186/s13059-019-1700-9" class="external-link">https://doi.org/10.1186/s13059-019-1700-9</a>.
</div>
<div id="ref-Petereit2016" class="csl-entry">
Petereit, Juli, Sebastian Smith, Frederick C. Harris, and Karen A.
Schlauch. 2016. <span>“<span class="nocase">petal: Co-expression network
modelling in R</span>.”</span> <em>BMC Systems Biology</em> 10 (S2): 51.
<a href="https://doi.org/10.1186/s12918-016-0298-8" class="external-link">https://doi.org/10.1186/s12918-016-0298-8</a>.
</div>
<div id="ref-Russo2018" class="csl-entry">
Russo, Pedro S. T., Gustavo R. Ferreira, Lucas E. Cardozo, Matheus C.
Bürger, Raul Arias-Carrasco, Sandra R. Maruyama, Thiago D. C. Hirata, et
al. 2018. <span>“<span class="nocase">CEMiTool: a Bioconductor package
for performing comprehensive modular co-expression
analyses</span>.”</span> <em>BMC Bioinformatics</em> 19 (1): 56. <a href="https://doi.org/10.1186/s12859-018-2053-1" class="external-link">https://doi.org/10.1186/s12859-018-2053-1</a>.
</div>
<div id="ref-Shin2020" class="csl-entry">
Shin, Junha, Harald Marx, Alicia Richards, Dries Vaneechoutte,
Dhileepkumar Jayaraman, Junko Maeda, Sanhita Chakraborty, et al. 2020.
<span>“<span class="nocase">A network-based comparative framework to
study conservation and divergence of proteomes in plant
phylogenies</span>.”</span> <em>Nucleic Acids Research</em>, 1–23. <a href="https://doi.org/10.1093/nar/gkaa1041" class="external-link">https://doi.org/10.1093/nar/gkaa1041</a>.
</div>
<div id="ref-VanBel2018" class="csl-entry">
Van Bel, Michiel, Tim Diels, Emmelien Vancaester, Lukasz Kreft,
Alexander Botzki, Yves Van De Peer, Frederik Coppens, and Klaas
Vandepoele. 2018. <span>“<span class="nocase">PLAZA 4.0: An integrative
resource for functional, evolutionary and comparative plant
genomics</span>.”</span> <em>Nucleic Acids Research</em> 46 (D1):
D1190–96. <a href="https://doi.org/10.1093/nar/gkx1002" class="external-link">https://doi.org/10.1093/nar/gkx1002</a>.
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p><strong>NOTE:</strong> In case you have many
tab-separated expression tables in a directory, <code>BioNERO</code> has
a helper function named <code><a href="../reference/dfs2one.html">dfs2one()</a></code> to load all these files
and combine them into a single data frame.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><strong>NOTE:</strong> Here, we are using TPM-normalized
data. If you have expression data as raw read counts, set the argument
<code>vstransform = TRUE</code> in <code><a href="../reference/exp_preprocess.html">exp_preprocess()</a></code>. This
will apply DESeq2’s variance stabilizing transformation <span class="citation">(Love, Huber, and Anders 2014)</span> to your count
data.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><strong>PRO TIP:</strong> If your <span class="math inline">\(\beta\)</span> power is too low (say below 6),
look at the plot as a sanity check.<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p><strong>NOTE:</strong> The calculations performed by
this function may take a long time depending on the your network size.
Use it only if you have some reason to suspect that the modules are
highly dependent on a particular set of samples.<a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p><strong>NOTE:</strong> The function
<code><a href="../reference/gene_significance.html">gene_significance()</a></code> works just like
<code><a href="../reference/module_trait_cor.html">module_trait_cor()</a></code>, but it correlates individual genes (not
the whole module) to traits. This function is very useful if you have a
set of candidate genes and you want to find which of them are more
associated with the trait of interest. See
<code>?gene_significance()</code> for more details.<a href="#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p><strong>NOTE:</strong> The functions
<code><a href="../reference/module_enrichment.html">module_enrichment()</a></code> and <code><a href="../reference/enrichment_analysis.html">enrichment_analysis()</a></code>
can be parallelized with <code>BiocParallel</code> to increase speed.
The default parallel back-end is <em>SerialParam()</em>, but this can be
modified in the argument <code>bp_param</code>.<a href="#fnref6" class="footnote-back">↩︎</a></p></li>
<li id="fn7"><p><strong>PRO TIP:</strong> Generally, we advise you to
filter by optimal scale-free topology fit (default). However, if you
want to specify your own correlation filter for some reason (e.g.,
visualization), we <strong>strongly recommend</strong> using the
function <code><a href="../reference/check_SFT.html">check_SFT()</a></code> to check if your resulting graph
satisfies the scale-free topology. If it does not, then your graph does
not resemble <em>real</em> biological networks and, hence, one cannot
trust it for biological interpretations.<a href="#fnref7" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Fabricio Almeida-Silva, Thiago Venancio.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
