<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="BioNERO">
<title>Gene regulatory network inference • BioNERO</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.2/font.css" rel="stylesheet">
<link href="../deps/Source_Code_Pro-0.4.2/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Gene regulatory network inference">
<meta property="og:description" content="BioNERO">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-#0092ac"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">BioNERO</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/vignette_01_GCN_inference.html">Gene coexpression network inference</a>
    <a class="dropdown-item" href="../articles/vignette_02_GRN_inference.html">Gene regulatory network inference</a>
    <a class="dropdown-item" href="../articles/vignette_03_network_comparison.html">Network comparison: consensus modules and module preservation</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/almeidasilvaf/BioNERO/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="vignette_02_GRN_inference_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="vignette_02_GRN_inference_files/d3-4.5.0/d3.min.js"></script><script src="vignette_02_GRN_inference_files/forceNetwork-binding-0.4/forceNetwork.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Gene regulatory network inference</h1>
                        <h4 data-toc-skip class="author">Fabricio
Almeida-Silva</h4>
            <address class="author_afil">
      Universidade Estadual do Norte Fluminense Darcy Ribeiro, RJ,
Brazil<br><h4 data-toc-skip class="author">Thiago
Motta Venancio</h4>
            <address class="author_afil">
      Universidade Estadual do Norte Fluminense Darcy Ribeiro, RJ,
Brazil<br><small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/BioNERO/blob/HEAD/vignettes/vignette_02_GRN_inference.Rmd" class="external-link"><code>vignettes/vignette_02_GRN_inference.Rmd</code></a></small>
      <div class="d-none name"><code>vignette_02_GRN_inference.Rmd</code></div>
    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">'BiocManager'</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"BioNERO"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load package after installation</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/BioNERO" class="external-link">BioNERO</a></span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co"># for reproducibility</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction-and-algorithm-description">Introduction and algorithm description<a class="anchor" aria-label="anchor" href="#introduction-and-algorithm-description"></a>
</h2>
<p>In the previous vignette, we explored all aspects of gene
coexpression networks (GCNs), which are represented as
<strong>undirected weighted graphs</strong>. It is
<strong>undirected</strong> because, for a given link between <em>gene
A</em> and <em>gene B</em>, we can only say that these genes are
coexpressed, but we cannot know whether <em>gene A</em> controls
<em>gene B</em> or otherwise. Further, <strong>weighted</strong> means
that some coexpression relationships between gene pairs are stronger
than others. In this vignette, we will demonstrate how to infer gene
regulatory networks (GRNs) from expression data with BioNERO. GRNs
display interactions between regulators (e.g., transcription factors or
miRNAs) and their targets (e.g., genes). Hence, they are represented as
<strong>directed unweighted graphs</strong>.</p>
<p>Numerous algorithms have been developed to infer GRNs from expression
data. However, the algorithm performances are highly dependent on the
benchmark data set. To solve this uncertainty, <span class="citation">Marbach et al. (2012)</span> proposed the application
of the <em>“wisdom of the crowds”</em> principle to GRN inference. This
approach consists in inferring GRNs with different algorithms, ranking
the interactions identified by each method, and calculating the average
rank for each interaction across all algorithms used. This way, we can
have consensus, high-confidence edges to be used in biological
interpretations. For that, <code>BioNERO</code> implements three popular
algorithms: GENIE3 <span class="citation">(Huynh-Thu et al.
2010)</span>, ARACNE <span class="citation">(Margolin et al.
2006)</span> and CLR <span class="citation">(Faith et al.
2007)</span>.</p>
</div>
<div class="section level2">
<h2 id="data-preprocessing">Data preprocessing<a class="anchor" aria-label="anchor" href="#data-preprocessing"></a>
</h2>
<p>Before inferring the GRN, we will preprocess the expression data the
same way we did in the previous vignette.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load example data set</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">zma.se</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Preprocess the expression data</span></span>
<span><span class="va">final_exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exp_preprocess.html">exp_preprocess</a></span><span class="op">(</span><span class="va">zma.se</span>, min_exp <span class="op">=</span> <span class="fl">10</span>, variance_filter <span class="op">=</span> <span class="cn">TRUE</span>, n<span class="op">=</span><span class="fl">2000</span><span class="op">)</span></span>
<span><span class="co">## Number of removed samples: 1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="gene-regulatory-network-inference">Gene regulatory network inference<a class="anchor" aria-label="anchor" href="#gene-regulatory-network-inference"></a>
</h2>
<p><code>BioNERO</code> requires only 2 objects for GRN inference: the
<strong>expression data</strong> (SummarizedExperiment, matrix or data
frame) and a character vector of <strong>regulators</strong>
(transcription factors or miRNAs). The transcription factors used in
this vignette were downloaded from PlantTFDB 4.0 <span class="citation">(Jin et al. 2017)</span>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">zma.tfs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">zma.tfs</span><span class="op">)</span></span>
<span><span class="co">##              Gene Family</span></span>
<span><span class="co">## 6  Zm00001d022525    Dof</span></span>
<span><span class="co">## 25 Zm00001d037605   GATA</span></span>
<span><span class="co">## 28 Zm00001d049540    NAC</span></span>
<span><span class="co">## 45 Zm00001d042287    MYB</span></span>
<span><span class="co">## 46 Zm00001d042288    NAC</span></span>
<span><span class="co">## 54 Zm00001d039371    TCP</span></span></code></pre></div>
<div class="section level3">
<h3 id="consensus-grn-inference">Consensus GRN inference<a class="anchor" aria-label="anchor" href="#consensus-grn-inference"></a>
</h3>
<p>Inferring GRNs based on the <em>wisdom of the crowds</em> principle
can be done with a single function: <code><a href="../reference/exp2grn.html">exp2grn()</a></code>. This
function will infer GRNs with GENIE3, ARACNE and CLR, calculate average
ranks for each interaction and filter the resulting network based on the
optimal scale-free topology (SFT) fit. In the filtering step, <em>n</em>
different networks are created by subsetting the top <em>n</em>
quantiles. For instance, if a network of 10,000 edges is given as input
with <code>nsplit = 10</code>, 10 different networks will be created:
the first with 1,000 edges, the second with 2,000 edges, and so on, with
the last network being the original input network. Then, for each
network, the function will calculate the SFT fit and select the best
fit.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Using 10 trees for demonstration purposes. Use the default: 1000</span></span>
<span><span class="va">grn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exp2grn.html">exp2grn</a></span><span class="op">(</span>exp <span class="op">=</span> <span class="va">final_exp</span>, regulators <span class="op">=</span> <span class="va">zma.tfs</span><span class="op">$</span><span class="va">Gene</span>, nTrees<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">## The top number of edges that best fits the scale-free topology is 247</span></span></code></pre></div>
<p><img src="vignette_02_GRN_inference_files/figure-html/exp2grn-1.png" width="480" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">grn</span><span class="op">)</span></span>
<span><span class="co">##          Regulator         Target</span></span>
<span><span class="co">## 290 Zm00001d041474 Zm00001d018986</span></span>
<span><span class="co">## 280 Zm00001d041474 Zm00001d006602</span></span>
<span><span class="co">## 281 Zm00001d041474 Zm00001d006942</span></span>
<span><span class="co">## 325 Zm00001d044315 Zm00001d043497</span></span>
<span><span class="co">## 65  Zm00001d013777 Zm00001d046996</span></span>
<span><span class="co">## 252 Zm00001d038832 Zm00001d021147</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="algorithm-specific-grn-inference">Algorithm-specific GRN inference<a class="anchor" aria-label="anchor" href="#algorithm-specific-grn-inference"></a>
</h3>
<p>This section is directed to users who, for some reason (e.g.,
comparison, exploration), want to infer GRNs with particular algorithms.
The available algorithms are:</p>
<p><strong>GENIE3:</strong> a regression-tree based algorithm that
decomposes the prediction of GRNs for <em>n</em> genes into <em>n</em>
regression problems. For each regression problem, the expression profile
of a target gene is predicted from the expression profiles of all other
genes using random forests (default) or extra-trees.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Using 10 trees for demonstration purposes. Use the default: 1000</span></span>
<span><span class="va">genie3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/grn_infer.html">grn_infer</a></span><span class="op">(</span><span class="va">final_exp</span>, method <span class="op">=</span> <span class="st">"genie3"</span>, regulators <span class="op">=</span> <span class="va">zma.tfs</span><span class="op">$</span><span class="va">Gene</span>, </span>
<span>                    nTrees<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genie3</span><span class="op">)</span></span>
<span><span class="co">##                Node1          Node2    Weight</span></span>
<span><span class="co">## 20352 Zm00001d041474 Zm00001d017881 0.5439514</span></span>
<span><span class="co">## 41340 Zm00001d034751 Zm00001d037111 0.5322394</span></span>
<span><span class="co">## 13037 Zm00001d034751 Zm00001d012407 0.4348469</span></span>
<span><span class="co">## 13207 Zm00001d045323 Zm00001d012513 0.4203583</span></span>
<span><span class="co">## 55378 Zm00001d028432 Zm00001d048693 0.4071160</span></span>
<span><span class="co">## 50200 Zm00001d013777 Zm00001d044212 0.3957483</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">genie3</span><span class="op">)</span></span>
<span><span class="co">## [1] 60136     3</span></span></code></pre></div>
<p><strong>ARACNE:</strong> information-theoretic algorithm that aims to
remove indirect interactions inferred by coexpression.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aracne</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/grn_infer.html">grn_infer</a></span><span class="op">(</span><span class="va">final_exp</span>, method <span class="op">=</span> <span class="st">"aracne"</span>, regulators <span class="op">=</span> <span class="va">zma.tfs</span><span class="op">$</span><span class="va">Gene</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">aracne</span><span class="op">)</span></span>
<span><span class="co">##                Node1          Node2   Weight</span></span>
<span><span class="co">## 23861 Zm00001d038832 Zm00001d021147 1.789818</span></span>
<span><span class="co">## 1758  Zm00001d038832 Zm00001d000432 1.692232</span></span>
<span><span class="co">## 11337 Zm00001d038832 Zm00001d011086 1.692232</span></span>
<span><span class="co">## 27014 Zm00001d011139 Zm00001d024274 1.674840</span></span>
<span><span class="co">## 51070 Zm00001d011139 Zm00001d045069 1.658043</span></span>
<span><span class="co">## 28387 Zm00001d038832 Zm00001d025784 1.641802</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">aracne</span><span class="op">)</span></span>
<span><span class="co">## [1] 411   3</span></span></code></pre></div>
<p><strong>CLR:</strong> extension of the relevance networks algorithm
that uses mutual information to identify regulatory interactions.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/grn_infer.html">grn_infer</a></span><span class="op">(</span><span class="va">final_exp</span>, method <span class="op">=</span> <span class="st">"clr"</span>, regulators <span class="op">=</span> <span class="va">zma.tfs</span><span class="op">$</span><span class="va">Gene</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clr</span><span class="op">)</span></span>
<span><span class="co">##                Node1          Node2   Weight</span></span>
<span><span class="co">## 26302 Zm00001d046937 Zm00001d023376 12.70216</span></span>
<span><span class="co">## 11267 Zm00001d046937 Zm00001d011080 12.25336</span></span>
<span><span class="co">## 12540 Zm00001d041474 Zm00001d012007 10.74023</span></span>
<span><span class="co">## 51019 Zm00001d042263 Zm00001d045042 10.50925</span></span>
<span><span class="co">## 17810 Zm00001d041474 Zm00001d015811 10.33216</span></span>
<span><span class="co">## 29278 Zm00001d046937 Zm00001d026632 10.20075</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">clr</span><span class="op">)</span></span>
<span><span class="co">## [1] 26657     3</span></span></code></pre></div>
<p>Users can also infer GRNs with the 3 algorithms at once using the
function <code>exp_combined()</code>. The resulting edge lists are
stored in a list of 3 elements. <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;strong&gt;NOTE:&lt;/strong&gt; Under the hood,
&lt;code&gt;&lt;a href="../reference/exp2grn.html"&gt;exp2grn()&lt;/a&gt;&lt;/code&gt; uses &lt;code&gt;exp_combined()&lt;/code&gt; followed by
averaging ranks with &lt;code&gt;&lt;a href="../reference/grn_average_rank.html"&gt;grn_average_rank()&lt;/a&gt;&lt;/code&gt; and filtering with
&lt;code&gt;&lt;a href="../reference/grn_filter.html"&gt;grn_filter()&lt;/a&gt;&lt;/code&gt;.&lt;/p&gt;'><sup>1</sup></a></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">grn_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/grn_combined.html">grn_combined</a></span><span class="op">(</span><span class="va">final_exp</span>, regulators <span class="op">=</span> <span class="va">zma.tfs</span><span class="op">$</span><span class="va">Gene</span>, nTrees<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">grn_list</span><span class="op">$</span><span class="va">genie3</span><span class="op">)</span></span>
<span><span class="co">##                Node1          Node2    Weight</span></span>
<span><span class="co">## 12013 Zm00001d041474 Zm00001d011541 0.4629469</span></span>
<span><span class="co">## 30418 Zm00001d046568 Zm00001d027841 0.4289222</span></span>
<span><span class="co">## 33403 Zm00001d041474 Zm00001d030748 0.4140894</span></span>
<span><span class="co">## 6910  Zm00001d044315 Zm00001d006725 0.4103733</span></span>
<span><span class="co">## 22057 Zm00001d041474 Zm00001d018986 0.4020641</span></span>
<span><span class="co">## 45153 Zm00001d034751 Zm00001d039733 0.3935705</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">grn_list</span><span class="op">$</span><span class="va">aracne</span><span class="op">)</span></span>
<span><span class="co">##                Node1          Node2   Weight</span></span>
<span><span class="co">## 23861 Zm00001d038832 Zm00001d021147 1.789818</span></span>
<span><span class="co">## 1758  Zm00001d038832 Zm00001d000432 1.692232</span></span>
<span><span class="co">## 11337 Zm00001d038832 Zm00001d011086 1.692232</span></span>
<span><span class="co">## 27014 Zm00001d011139 Zm00001d024274 1.674840</span></span>
<span><span class="co">## 51070 Zm00001d011139 Zm00001d045069 1.658043</span></span>
<span><span class="co">## 28387 Zm00001d038832 Zm00001d025784 1.641802</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">grn_list</span><span class="op">$</span><span class="va">clr</span><span class="op">)</span></span>
<span><span class="co">##                Node1          Node2   Weight</span></span>
<span><span class="co">## 26302 Zm00001d046937 Zm00001d023376 12.70216</span></span>
<span><span class="co">## 11267 Zm00001d046937 Zm00001d011080 12.25336</span></span>
<span><span class="co">## 12540 Zm00001d041474 Zm00001d012007 10.74023</span></span>
<span><span class="co">## 51019 Zm00001d042263 Zm00001d045042 10.50925</span></span>
<span><span class="co">## 17810 Zm00001d041474 Zm00001d015811 10.33216</span></span>
<span><span class="co">## 29278 Zm00001d046937 Zm00001d026632 10.20075</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="gene-regulatory-network-analysis">Gene regulatory network analysis<a class="anchor" aria-label="anchor" href="#gene-regulatory-network-analysis"></a>
</h2>
<p>After inferring the GRN, <code>BioNERO</code> allows users to perform
some common downstream analyses.</p>
<div class="section level3">
<h3 id="hub-gene-identification">Hub gene identification<a class="anchor" aria-label="anchor" href="#hub-gene-identification"></a>
</h3>
<p>GRN hubs are defined as the top 10% most highly connected regulators,
but this percentile is flexible in <code>BioNERO</code>.<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;&lt;strong&gt;NOTE:&lt;/strong&gt; Remember: GRNs are represented as
&lt;strong&gt;directed&lt;/strong&gt; graphs. This implies that only regulators are
taken into account when identifying hubs. The goal here is to identify
regulators (e.g., transcription factors) that control the expression of
several genes.&lt;/p&gt;"><sup>2</sup></a> They can be identified
with <code><a href="../reference/get_hubs_grn.html">get_hubs_grn()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hubs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_hubs_grn.html">get_hubs_grn</a></span><span class="op">(</span><span class="va">grn</span><span class="op">)</span></span>
<span><span class="va">hubs</span></span>
<span><span class="co">##              Gene Degree</span></span>
<span><span class="co">## 1  Zm00001d038832     16</span></span>
<span><span class="co">## 2  Zm00001d041474     13</span></span>
<span><span class="co">## 3  Zm00001d046937     13</span></span>
<span><span class="co">## 4  Zm00001d011139     12</span></span>
<span><span class="co">## 5  Zm00001d052229     11</span></span>
<span><span class="co">## 6  Zm00001d013777     10</span></span>
<span><span class="co">## 7  Zm00001d039989     10</span></span>
<span><span class="co">## 8  Zm00001d038227     10</span></span>
<span><span class="co">## 9  Zm00001d030617     10</span></span>
<span><span class="co">## 10 Zm00001d044315      9</span></span>
<span><span class="co">## 11 Zm00001d003822      9</span></span>
<span><span class="co">## 12 Zm00001d020020      9</span></span>
<span><span class="co">## 13 Zm00001d046568      9</span></span>
<span><span class="co">## 14 Zm00001d010227      9</span></span>
<span><span class="co">## 15 Zm00001d025339      8</span></span>
<span><span class="co">## 16 Zm00001d028974      8</span></span>
<span><span class="co">## 17 Zm00001d042267      7</span></span>
<span><span class="co">## 18 Zm00001d014377      7</span></span>
<span><span class="co">## 19 Zm00001d054038      6</span></span>
<span><span class="co">## 20 Zm00001d042263      6</span></span>
<span><span class="co">## 21 Zm00001d035440      6</span></span>
<span><span class="co">## 22 Zm00001d036148      6</span></span>
<span><span class="co">## 23 Zm00001d031655      6</span></span>
<span><span class="co">## 24 Zm00001d034751      6</span></span>
<span><span class="co">## 25 Zm00001d018081      6</span></span>
<span><span class="co">## 26 Zm00001d027957      5</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="network-visualization">Network visualization<a class="anchor" aria-label="anchor" href="#network-visualization"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_grn.html">plot_grn</a></span><span class="op">(</span><span class="va">grn</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_02_GRN_inference_files/figure-html/plot_static-1.png" width="384" style="display: block; margin: auto;"></p>
<p>GRNs can also be visualized interactively for exploratory
purposes.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_grn.html">plot_grn</a></span><span class="op">(</span><span class="va">grn</span>, interactive <span class="op">=</span> <span class="cn">TRUE</span>, dim_interactive <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">500</span>,<span class="fl">500</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-82606d076d63ab4a598c" style="width:500px;height:500px;" class="forceNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-82606d076d63ab4a598c">{"x":{"links":{"source":[18,17,3,26,12,22,3,25,3,24,1,1,16,24,5,20,3,22,18,0,0,16,1,7,14,0,1,25,6,1,4,1,12,8,3,5,2,5,3,29,18,6,26,0,19,5,17,17,18,2,23,6,1,19,1,24,6,27,12,0,28,5,30,26,24,6,6,0,0,10,9,10,10,5,26,17,12,2,11,23,26,5,0,27,3,0,27,14,18,9,4,13,3,3,17,2,9,11,20,14,19,11,15,23,5,15,16,7,23,24,7,3,6,28,16,23,5,19,20,0,13,22,28,20,15,26,24,9,26,2,28,15,8,24,17,4,2,25,2,30,22,21,12,12,18,20,22,16,8,5,20,9,10,21,0,21,17,12,21,18,7,7,16,9,7,18,22,14,29,27,21,22,13,24,13,0,13,25,9,29,26,11,12,6,21,16,15,22,2,18,4,12,24,10,1,25,8,23,8,3,17,22,22,21,4,10,7,3,17,26,8,5,2,30,4,16,7,7,23,9,3,7,8,15,11,16,3,14,30,19,3,6,14,6,23,0,7,5,25,5,2,7,26,24,11,3,6],"target":[252,164,108,250,113,212,152,77,161,75,145,197,192,209,211,200,136,151,61,32,33,55,129,242,175,50,114,96,96,187,58,54,204,41,150,135,153,38,239,254,257,147,183,89,120,112,92,109,185,98,70,244,190,88,207,243,233,233,47,43,130,166,166,165,165,255,248,191,221,139,103,103,95,196,127,202,80,195,46,186,111,216,68,213,177,31,227,251,106,132,102,49,36,259,176,170,105,107,179,51,256,168,53,53,232,234,240,84,181,91,48,48,140,203,73,73,220,188,123,57,90,124,156,66,97,149,149,64,249,229,157,74,52,184,184,122,56,174,169,169,69,208,86,87,180,206,253,238,133,261,160,172,225,141,142,215,76,171,137,167,119,235,173,237,60,121,78,231,258,85,194,138,158,144,128,82,155,198,42,182,116,228,99,214,67,65,65,217,199,230,125,218,146,44,34,223,118,118,201,154,205,126,134,222,37,193,40,93,59,79,236,224,35,241,131,246,260,104,81,189,178,162,226,115,115,115,143,110,245,62,94,83,72,71,100,45,219,247,159,63,148,117,163,163,210,101,39],"colour":["#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666","#666"]},"nodes":{"name":["Zm00001d041474","Zm00001d044315","Zm00001d013777","Zm00001d038832","Zm00001d054038","Zm00001d046937","Zm00001d052229","Zm00001d011139","Zm00001d042267","Zm00001d025339","Zm00001d042263","Zm00001d035440","Zm00001d003822","Zm00001d027957","Zm00001d036148","Zm00001d031655","Zm00001d020020","Zm00001d046568","Zm00001d010227","Zm00001d018208","Zm00001d034751","Zm00001d014377","Zm00001d039989","Zm00001d028974","Zm00001d038227","Zm00001d018081","Zm00001d030617","Zm00001d028432","Zm00001d002654","Zm00001d045323","Zm00001d045044","Zm00001d018986","Zm00001d006602","Zm00001d006942","Zm00001d043497","Zm00001d046996","Zm00001d021147","Zm00001d045041","Zm00001d011080","Zm00001d054093","Zm00001d045069","Zm00001d010618","Zm00001d040735","Zm00001d014773","Zm00001d043382","Zm00001d051676","Zm00001d018192","Zm00001d014519","Zm00001d025784","Zm00001d020926","Zm00001d008691","Zm00001d022237","Zm00001d032442","Zm00001d022569","Zm00001d009774","Zm00001d007097","Zm00001d032668","Zm00001d027374","Zm00001d009418","Zm00001d046492","Zm00001d037877","Zm00001d006474","Zm00001d049826","Zm00001d052122","Zm00001d029968","Zm00001d042245","Zm00001d028998","Zm00001d042216","Zm00001d018668","Zm00001d033301","Zm00001d013319","Zm00001d051272","Zm00001d049965","Zm00001d026490","Zm00001d031349","Zm00001d003817","Zm00001d036738","Zm00001d003277","Zm00001d037962","Zm00001d046593","Zm00001d017169","Zm00001d047913","Zm00001d040477","Zm00001d049867","Zm00001d024274","Zm00001d038161","Zm00001d033840","Zm00001d033846","Zm00001d014073","Zm00001d011541","Zm00001d027652","Zm00001d024518","Zm00001d012339","Zm00001d046317","Zm00001d049835","Zm00001d016572","Zm00001d008962","Zm00001d029151","Zm00001d013300","Zm00001d042025","Zm00001d051420","Zm00001d054017","Zm00001d020381","Zm00001d016568","Zm00001d047498","Zm00001d021709","Zm00001d020108","Zm00001d021773","Zm00001d000432","Zm00001d012597","Zm00001d048634","Zm00001d018366","Zm00001d011855","Zm00001d001881","Zm00001d008826","Zm00001d048431","Zm00001d041794","Zm00001d053625","Zm00001d043693","Zm00001d037489","Zm00001d011816","Zm00001d037937","Zm00001d032547","Zm00001d027342","Zm00001d027731","Zm00001d042481","Zm00001d044575","Zm00001d016926","Zm00001d040323","Zm00001d007422","Zm00001d014785","Zm00001d047448","Zm00001d020236","Zm00001d034644","Zm00001d044747","Zm00001d010814","Zm00001d006008","Zm00001d037257","Zm00001d039043","Zm00001d016567","Zm00001d025892","Zm00001d035562","Zm00001d035664","Zm00001d048609","Zm00001d039131","Zm00001d004125","Zm00001d043217","Zm00001d011401","Zm00001d053197","Zm00001d029965","Zm00001d010733","Zm00001d006160","Zm00001d003127","Zm00001d011004","Zm00001d044035","Zm00001d040508","Zm00001d028201","Zm00001d031168","Zm00001d039081","Zm00001d052093","Zm00001d035054","Zm00001d003539","Zm00001d048163","Zm00001d053663","Zm00001d000417","Zm00001d015004","Zm00001d014968","Zm00001d037346","Zm00001d022546","Zm00001d033128","Zm00001d021508","Zm00001d037052","Zm00001d035115","Zm00001d037684","Zm00001d032830","Zm00001d008610","Zm00001d021301","Zm00001d018896","Zm00001d048157","Zm00001d022134","Zm00001d033910","Zm00001d024379","Zm00001d041593","Zm00001d011500","Zm00001d032517","Zm00001d013141","Zm00001d018359","Zm00001d009339","Zm00001d027288","Zm00001d048085","Zm00001d013757","Zm00001d015279","Zm00001d004936","Zm00001d045042","Zm00001d038547","Zm00001d017719","Zm00001d016720","Zm00001d004821","Zm00001d040561","Zm00001d042367","Zm00001d005791","Zm00001d043905","Zm00001d017072","Zm00001d026056","Zm00001d010455","Zm00001d044525","Zm00001d033979","Zm00001d014124","Zm00001d033559","Zm00001d005347","Zm00001d053671","Zm00001d005675","Zm00001d001913","Zm00001d018744","Zm00001d042030","Zm00001d036293","Zm00001d018639","Zm00001d042309","Zm00001d042613","Zm00001d051929","Zm00001d026632","Zm00001d015811","Zm00001d044939","Zm00001d043613","Zm00001d046952","Zm00001d035308","Zm00001d048165","Zm00001d019002","Zm00001d041809","Zm00001d030149","Zm00001d042476","Zm00001d037984","Zm00001d023376","Zm00001d014349","Zm00001d023694","Zm00001d037507","Zm00001d046661","Zm00001d037695","Zm00001d034558","Zm00001d011086","Zm00001d023987","Zm00001d047020","Zm00001d008257","Zm00001d014294","Zm00001d013320","Zm00001d049619","Zm00001d047470","Zm00001d052079","Zm00001d015138","Zm00001d030048","Zm00001d001857","Zm00001d019670","Zm00001d000123","Zm00001d034505","Zm00001d011087","Zm00001d015084","Zm00001d022301","Zm00001d011162","Zm00001d038057","Zm00001d021196","Zm00001d047481","Zm00001d034668"],"group":["Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Regulator","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target","Target"],"nodesize":[16,13,13,12,11,10,10,10,10,9,9,9,9,9,8,8,7,7,6,6,6,6,6,6,6,5,5,4,4,4,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]},"options":{"NodeID":"name","Group":"group","colourScale":"d3.scaleOrdinal() .domain([\"Regulator\", \"Target\"]) .range([\"forestgreen\", \"orange\"])","fontSize":20,"fontFamily":"serif","clickTextSize":50,"linkDistance":50,"linkWidth":"function(d) { return Math.sqrt(d.value); }","charge":-30,"opacity":1,"zoom":true,"legend":true,"arrows":false,"nodesize":true,"radiusCalculation":" Math.sqrt(d.nodesize)+6","bounded":false,"opacityNoHover":0,"clickAction":null}},"evals":[],"jsHooks":[]}</script><p>Finally, <code>BioNERO</code> can also be used for visualization and
hub identification in protein-protein (PPI) interaction networks. The
functions <code><a href="../reference/get_hubs_grn.html">get_hubs_ppi()</a></code> and <code><a href="../reference/plot_ppi.html">plot_ppi()</a></code> work
the same way as their equivalents for GRNs (<code><a href="../reference/get_hubs_grn.html">get_hubs_grn()</a></code>
and <code><a href="../reference/plot_grn.html">plot_grn()</a></code>).</p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This vignette was created under the following conditions:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] BioNERO_1.5.1    BiocStyle_2.24.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] backports_1.4.1             circlize_0.4.15            </span></span>
<span><span class="co">##   [3] Hmisc_4.7-1                 systemfonts_1.0.4          </span></span>
<span><span class="co">##   [5] plyr_1.8.7                  igraph_1.3.4               </span></span>
<span><span class="co">##   [7] splines_4.2.1               GENIE3_1.18.0              </span></span>
<span><span class="co">##   [9] BiocParallel_1.30.3         GenomeInfoDb_1.32.3        </span></span>
<span><span class="co">##  [11] ggnetwork_0.5.10            ggplot2_3.3.6              </span></span>
<span><span class="co">##  [13] sva_3.44.0                  digest_0.6.29              </span></span>
<span><span class="co">##  [15] foreach_1.5.2               htmltools_0.5.3            </span></span>
<span><span class="co">##  [17] GO.db_3.15.0                fansi_1.0.3                </span></span>
<span><span class="co">##  [19] magrittr_2.0.3              checkmate_2.1.0            </span></span>
<span><span class="co">##  [21] memoise_2.0.1               cluster_2.1.4              </span></span>
<span><span class="co">##  [23] doParallel_1.0.17           limma_3.52.2               </span></span>
<span><span class="co">##  [25] annotate_1.74.0             ComplexHeatmap_2.12.1      </span></span>
<span><span class="co">##  [27] fastcluster_1.2.3           Biostrings_2.64.1          </span></span>
<span><span class="co">##  [29] matrixStats_0.62.0          pkgdown_2.0.6              </span></span>
<span><span class="co">##  [31] jpeg_0.1-9                  colorspace_2.0-3           </span></span>
<span><span class="co">##  [33] blob_1.2.3                  ggrepel_0.9.1              </span></span>
<span><span class="co">##  [35] textshaping_0.3.6           xfun_0.32                  </span></span>
<span><span class="co">##  [37] dplyr_1.0.9                 crayon_1.5.1               </span></span>
<span><span class="co">##  [39] RCurl_1.98-1.8              jsonlite_1.8.0             </span></span>
<span><span class="co">##  [41] genefilter_1.78.0           impute_1.70.0              </span></span>
<span><span class="co">##  [43] survival_3.4-0              iterators_1.0.14           </span></span>
<span><span class="co">##  [45] glue_1.6.2                  gtable_0.3.0               </span></span>
<span><span class="co">##  [47] zlibbioc_1.42.0             XVector_0.36.0             </span></span>
<span><span class="co">##  [49] GetoptLong_1.0.5            DelayedArray_0.22.0        </span></span>
<span><span class="co">##  [51] shape_1.4.6                 BiocGenerics_0.42.0        </span></span>
<span><span class="co">##  [53] abind_1.4-5                 scales_1.2.1               </span></span>
<span><span class="co">##  [55] edgeR_3.38.4                DBI_1.1.3                  </span></span>
<span><span class="co">##  [57] Rcpp_1.0.9                  xtable_1.8-4               </span></span>
<span><span class="co">##  [59] htmlTable_2.4.1             clue_0.3-61                </span></span>
<span><span class="co">##  [61] foreign_0.8-82              bit_4.0.4                  </span></span>
<span><span class="co">##  [63] preprocessCore_1.58.0       Formula_1.2-4              </span></span>
<span><span class="co">##  [65] stats4_4.2.1                htmlwidgets_1.5.4          </span></span>
<span><span class="co">##  [67] httr_1.4.4                  RColorBrewer_1.1-3         </span></span>
<span><span class="co">##  [69] farver_2.1.1                XML_3.99-0.10              </span></span>
<span><span class="co">##  [71] pkgconfig_2.0.3             nnet_7.3-17                </span></span>
<span><span class="co">##  [73] sass_0.4.2                  deldir_1.0-6               </span></span>
<span><span class="co">##  [75] locfit_1.5-9.6              utf8_1.2.2                 </span></span>
<span><span class="co">##  [77] dynamicTreeCut_1.63-1       labeling_0.4.2             </span></span>
<span><span class="co">##  [79] reshape2_1.4.4              tidyselect_1.1.2           </span></span>
<span><span class="co">##  [81] rlang_1.0.4                 AnnotationDbi_1.58.0       </span></span>
<span><span class="co">##  [83] munsell_0.5.0               tools_4.2.1                </span></span>
<span><span class="co">##  [85] cachem_1.0.6                cli_3.3.0                  </span></span>
<span><span class="co">##  [87] generics_0.1.3              RSQLite_2.2.16             </span></span>
<span><span class="co">##  [89] statnet.common_4.6.0        evaluate_0.16              </span></span>
<span><span class="co">##  [91] stringr_1.4.1               fastmap_1.1.0              </span></span>
<span><span class="co">##  [93] yaml_2.3.5                  ragg_1.2.2                 </span></span>
<span><span class="co">##  [95] RhpcBLASctl_0.21-247.1      knitr_1.40                 </span></span>
<span><span class="co">##  [97] bit64_4.0.5                 fs_1.5.2                   </span></span>
<span><span class="co">##  [99] purrr_0.3.4                 KEGGREST_1.36.3            </span></span>
<span><span class="co">## [101] nlme_3.1-159                compiler_4.2.1             </span></span>
<span><span class="co">## [103] rstudioapi_0.14             png_0.1-7                  </span></span>
<span><span class="co">## [105] minet_3.54.0                tibble_3.1.8               </span></span>
<span><span class="co">## [107] statmod_1.4.37              bslib_0.4.0                </span></span>
<span><span class="co">## [109] stringi_1.7.8               highr_0.9                  </span></span>
<span><span class="co">## [111] desc_1.4.1                  lattice_0.20-45            </span></span>
<span><span class="co">## [113] Matrix_1.4-1                vctrs_0.4.1                </span></span>
<span><span class="co">## [115] pillar_1.8.1                lifecycle_1.0.1            </span></span>
<span><span class="co">## [117] networkD3_0.4               BiocManager_1.30.18        </span></span>
<span><span class="co">## [119] jquerylib_0.1.4             GlobalOptions_0.1.2        </span></span>
<span><span class="co">## [121] data.table_1.14.2           bitops_1.0-7               </span></span>
<span><span class="co">## [123] patchwork_1.1.2             GenomicRanges_1.48.0       </span></span>
<span><span class="co">## [125] R6_2.5.1                    latticeExtra_0.6-30        </span></span>
<span><span class="co">## [127] bookdown_0.28               network_1.17.2             </span></span>
<span><span class="co">## [129] gridExtra_2.3               IRanges_2.30.1             </span></span>
<span><span class="co">## [131] codetools_0.2-18            SummarizedExperiment_1.26.1</span></span>
<span><span class="co">## [133] rprojroot_2.0.3             rjson_0.2.21               </span></span>
<span><span class="co">## [135] S4Vectors_0.34.0            GenomeInfoDbData_1.2.8     </span></span>
<span><span class="co">## [137] intergraph_2.0-2            mgcv_1.8-40                </span></span>
<span><span class="co">## [139] parallel_4.2.1              grid_4.2.1                 </span></span>
<span><span class="co">## [141] rpart_4.1.16                NetRep_1.2.4               </span></span>
<span><span class="co">## [143] coda_0.19-4                 rmarkdown_2.16             </span></span>
<span><span class="co">## [145] MatrixGenerics_1.8.1        ggnewscale_0.4.7           </span></span>
<span><span class="co">## [147] Biobase_2.56.0              WGCNA_1.71                 </span></span>
<span><span class="co">## [149] base64enc_0.1-3             interp_1.1-3</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Faith2007" class="csl-entry">
Faith, Jeremiah J., Boris Hayete, Joshua T. Thaden, Ilaria Mogno, Jamey
Wierzbowski, Guillaume Cottarel, Simon Kasif, James J. Collins, and
Timothy S. Gardner. 2007. <span>“<span class="nocase">Large-scale
mapping and validation of Escherichia coli transcriptional regulation
from a compendium of expression profiles</span>.”</span> <em>PLoS
Biology</em> 5 (1): 0054–66. <a href="https://doi.org/10.1371/journal.pbio.0050008" class="external-link">https://doi.org/10.1371/journal.pbio.0050008</a>.
</div>
<div id="ref-Huynh-Thu2010" class="csl-entry">
Huynh-Thu, Vân Anh, Alexandre Irrthum, Louis Wehenkel, and Pierre
Geurts. 2010. <span>“<span class="nocase">Inferring regulatory networks
from expression data using tree-based methods</span>.”</span> <em>PLoS
ONE</em> 5 (9): 1–10. <a href="https://doi.org/10.1371/journal.pone.0012776" class="external-link">https://doi.org/10.1371/journal.pone.0012776</a>.
</div>
<div id="ref-Jin2017" class="csl-entry">
Jin, J, F Tian, D C Yang, Y Q Meng, L Kong, J Luo, and G Gao. 2017.
<span>“<span class="nocase">PlantTFDB 4.0: toward a central hub for
transcription factors and regulatory interactions in
plants</span>.”</span> <em>Nucleic Acids Res</em> 45 (D1): D1040–45. <a href="https://doi.org/10.1093/nar/gkw982" class="external-link">https://doi.org/10.1093/nar/gkw982</a>.
</div>
<div id="ref-Marbach2012" class="csl-entry">
Marbach, Daniel, James C. Costello, Robert Küffner, Nicole M. Vega,
Robert J. Prill, Diogo M. Camacho, Kyle R. Allison, et al. 2012.
<span>“<span class="nocase">Wisdom of crowds for robust gene network
inference</span>.”</span> <em>Nature Methods</em> 9 (8): 796–804. <a href="https://doi.org/10.1038/nmeth.2016" class="external-link">https://doi.org/10.1038/nmeth.2016</a>.
</div>
<div id="ref-Margolin2006" class="csl-entry">
Margolin, Adam A., Ilya Nemenman, Katia Basso, Chris Wiggins, Gustavo
Stolovitzky, Riccardo Dalla Favera, and Andrea Califano. 2006.
<span>“<span class="nocase">ARACNE: An algorithm for the reconstruction
of gene regulatory networks in a mammalian cellular
context</span>.”</span> <em>BMC Bioinformatics</em> 7 (SUPPL.1): 1–15.
<a href="https://doi.org/10.1186/1471-2105-7-S1-S7" class="external-link">https://doi.org/10.1186/1471-2105-7-S1-S7</a>.
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fabricio Almeida-Silva, Thiago Venancio.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
